<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Retrieves Gene length data — getlength • goseq</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Retrieves Gene length data — getlength"><meta name="description" content="Gets the length of each gene in a vector."><meta property="og:description" content="Gets the length of each gene in a vector."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">
    <a class="navbar-brand me-2" href="../index.html">goseq</a>
    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.57.2</small>
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>
    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/federicomarini/goseq/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <h1>Retrieves Gene length data</h1>
      <small class="dont-index">Source: <a href="https://github.com/federicomarini/goseq/blob/devel/R/getlength.R" class="external-link"><code>R/getlength.R</code></a></small>
      <div class="d-none name"><code>getlength.Rd</code></div>
    </div>
    <div class="ref-description section level2">
    <p>Gets the length of each gene in a vector.</p>
    </div>
    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">getlength</span><span class="op">(</span><span class="va">genes</span>, <span class="va">genome</span>, <span class="va">id</span><span class="op">)</span></span></code></pre></div>
    </div>
    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
<dl><dt id="arg-genes">genes<a class="anchor" aria-label="anchor" href="#arg-genes"></a></dt>
<dd><p>A vector or list of the genes for which length information is
required.</p></dd>
<dt id="arg-genome">genome<a class="anchor" aria-label="anchor" href="#arg-genome"></a></dt>
<dd><p>A string identifying the genome that <code>genes</code> refer to.
For a list of supported organisms run <code>supportedGenomes</code>.</p></dd>
<dt id="arg-id">id<a class="anchor" aria-label="anchor" href="#arg-id"></a></dt>
<dd><p>A string identifying the gene identifier used by <code>genes</code>.
For a list of supported gene IDs run <code>supportedGeneIDs</code>.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>Returns a vector of the gene lengths, in the same order as
<code>genes</code>.  If length data is unavailable for a particular gene NA is
returned in that position.  The returned vector is intended for use with the
<code>bias.data</code> option of the <code><a href="nullp.html">nullp</a></code> function.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Length data is obtained from data obtained from the UCSC genome browser for
each combination of <code>genome</code> and <code>id</code>.  As fetching this data at
runtime is time consuming, a local copy of the length information for common
genomes and gene ID are included in the <span class="pkg">geneLenDataBase</span> package.  This
function uses this package to fetch the required data.</p>
<p>The length of a gene is taken to be the median length of all its mature,
mRNA, transcripts.  It is always preferable to obtain length information
directly for the gene ID used to summarize your count data, rather than
converting IDs and then using the supplied databases.  Even when two genes
have a one-to-one mapping between different identifier conventions (which is
often not the case), they frequently refer to slightly different regions of
the genome with different lengths.  It is therefore recommended that the
user perform the full analysis in terms of only one gene ID, or manually
obtain their own length data for the identifier used to bin reads by gene.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code>supportedGenomes</code>, <code>supportedGeneIDs</code>,
<code><a href="nullp.html">nullp</a></code>, <span class="pkg">geneLenDataBase</span></p></div>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Matthew D. Young <a href="mailto:myoung@wehi.edu.au">myoung@wehi.edu.au</a></p>
    </div>
    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ENSG00000124208"</span>, </span></span>
<span class="r-in"><span>           <span class="st">"ENSG00000182463"</span>, </span></span>
<span class="r-in"><span>           <span class="st">"ENSG00000124201"</span>, </span></span>
<span class="r-in"><span>           <span class="st">"ENSG00000124205"</span>, </span></span>
<span class="r-in"><span>           <span class="st">"ENSG00000124207"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">getlength</span><span class="op">(</span><span class="va">genes</span>,<span class="st">'hg19'</span>,<span class="st">'ensGene'</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading hg19 length data...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 1978.0 3133.0 2973.0 2593.0 3036.5</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>
    <footer><div class="pkgdown-footer-left">
  <p>Developed by Matthew Young, Nadia Davidson, Federico Marini.</p>
</div>
<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.9000.</p>
</div>
    </footer></div>
  </body></html>

