<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Does selection-unbiased testing for category enrichment amongst
differentially expressed (DE) genes for RNA-seq data.  By default, tests
gene ontology (GO) categories, but any categories may be tested."><title>goseq Gene Ontology analyser — goseq • goseq</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="goseq Gene Ontology analyser — goseq"><meta property="og:description" content="Does selection-unbiased testing for category enrichment amongst
differentially expressed (DE) genes for RNA-seq data.  By default, tests
gene ontology (GO) categories, but any categories may be tested."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">goseq</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.57.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/federicomarini/goseq/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>goseq Gene Ontology analyser</h1>
      <small class="dont-index">Source: <a href="https://github.com/federicomarini/goseq/blob/HEAD/R/goseq.R" class="external-link"><code>R/goseq.R</code></a></small>
      <div class="d-none name"><code>goseq.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Does selection-unbiased testing for category enrichment amongst
differentially expressed (DE) genes for RNA-seq data.  By default, tests
gene ontology (GO) categories, but any categories may be tested.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">goseq</span><span class="op">(</span></span>
<span>  <span class="va">pwf</span>,</span>
<span>  <span class="va">genome</span>,</span>
<span>  <span class="va">id</span>,</span>
<span>  gene2cat <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  test.cats <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GO:CC"</span>, <span class="st">"GO:BP"</span>, <span class="st">"GO:MF"</span><span class="op">)</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"Wallenius"</span>,</span>
<span>  repcnt <span class="op">=</span> <span class="fl">2000</span>,</span>
<span>  use_genes_without_cat <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>pwf</dt>
<dd><p>An object containing gene names, DE calls, the probability
weighting function. Usually generated by <code><a href="nullp.html">nullp</a></code>.</p></dd>


<dt>genome</dt>
<dd><p>A string identifying the genome that <code>genes</code> refer to.
For a list of supported organisms run <code>supportedGenomes</code>.</p></dd>


<dt>id</dt>
<dd><p>A string identifying the gene identifier used by <code>genes</code>.
For a list of supported gene IDs run <code>supportedGeneIDs</code>.</p></dd>


<dt>gene2cat</dt>
<dd><p>A data frame with two columns containing the mapping between
genes and the categories of interest.  Alternatively, a list where the names
are genes and each entry is a vector containing GO categories associated
with that gene (this is the output produced by <code><a href="getgo.html">getgo</a></code>).  If set
to <code>NULL</code> goseq attempts to fetch GO categories automatically using
<code><a href="getgo.html">getgo</a></code>.</p></dd>


<dt>test.cats</dt>
<dd><p>A vector specifying which categories to test for over
representation amongst DE genes.  See details for allowed options.</p></dd>


<dt>method</dt>
<dd><p>The method to use to calculate the unbiased category
enrichment scores.  Valid options are "Wallenius", "Sampling" &amp;
"Hypergeometric".  "Hypergeometric" and "Sampling" should almost never be
used (see details).</p></dd>


<dt>repcnt</dt>
<dd><p>Number of random samples to be calculated when random sampling
is used.  Ignored unless <code>method="Sampling"</code>.</p></dd>


<dt>use_genes_without_cat</dt>
<dd><p>A boolean to indicate whether genes without a
category should still be used. For example, a large number of gene may have
no GO term annotated. If this option is set to FALSE, those genes will be
ignored in the calculation of p-values (default behaviour). If this option
is set to TRUE, then these genes will count towards the total number of
genes outside the category being tested (default behaviour prior to version
1.15.2).</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>goseq returns a data frame with several columns.  The first column
gives the name of the category, the second gives the p-value for the
associated category being over represented amongst DE genes.  The third
column gives the p-value for the associated category being under represented
amongst DE genes.  The p-values have not been corrected for multiple
hypothesis testing. The fourth and fifth columns give the number of
differentially expressed genes in the category and total genes in the
category respectively. If any of the categories was a GO term, there will be
two additional columns for the GO term and its ontology.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The <code>pwf</code> argument is almost always the output of the function
<code><a href="nullp.html">nullp</a></code>.  This is a data frame with 3 columns, named "DEgenes",
"bias.data" and "pwf" with the rownames set to the gene names.  Each row
corresponds to a gene with the DEgenes column specifying if the gene is DE
(1 for DE, 0 for not DE), the bias.data column giving the numeric value of
the DE bias being accounted for (usually the gene length or number of
counts) and the pwf column giving the genes value on the probability
weighting function.</p>
<p><code>goseq</code> obtains length data from UCSC and GO mappings from the organism
packages (see <code>link{getgo}</code> and <code><a href="getlength.html">getlength</a></code> for details).
If your data is in an unsupported format you will need to obtain the GO
category mapping and supply them to the <code>goseq</code> function using the
<code>gene2cat</code> argument.</p>
<p>To use your own gene to category mapping with <code>goseq</code>, use the
<code>gene2cat</code> argument.  This argument takes a data.frame, with one column
containing gene IDs and the other containing the associated categories.  As
the mapping from gene &lt;-&gt; category is in general many to many there will be
multiple rows containing the same gene identifier.  Alternatively,
<code>gene2cat</code> can take a list, where the names are the genes and the
entries are the GO categories associated with the genes.  This is the format
produced by the <code><a href="getgo.html">getgo</a></code> function and is more space efficient
than the data.frame representation.</p>
<p>If <code>gene2cat</code> is left as <code>NULL</code>, <code>goseq</code> attempts to use
<code><a href="getgo.html">getgo</a></code> to fetch GO category to gene identifier mappings.</p>
<p>The PWF is usually calculated using the <code><a href="nullp.html">nullp</a></code> function to
correct for length bias. However, <code>goseq</code> will work with any vector of
weights.  Any bias can be accounted for so long as a weight for each gene is
supplied using this argument.  <code>NA</code>s are allowed in the "pwf" and
"bias.data" columns of the PWF data frame (these usually occur as a result
of missing length data for some genes).  Any entry which is <code>NA</code> is set
to the weighting of the median gene.</p>
<p>Valid options for the <code>test.cats</code> argument are any combination of
"GO:CC", "GO:BP", "GO:MF" &amp; "KEGG".  The three GO terms refer to the
Cellular Component, Biological Process and Molecular Function respectively.
"KEGG" refers to KEGG pathways.</p>
<p>The three methods, "Wallenius", "Sampling" &amp; "Hypergeometric", calculate the
p-values as follows.</p>
<p>"Wallenius" approximates the true distribution of numbers of members of a
category amongst DE genes by the Wallenius non-central hypergeometric
distribution.  This distribution assumes that within a category all genes
have the same probability of being chosen.  Therefore, this approximation
works best when the range in probabilities obtained by the probability
weighting function is small. "Wallenius" is the recommended method for
calculating p-values.</p>
<p>"Sampling" uses random sampling to approximate the true distribution and
uses it to calculate the p-values for over (and under) representation of
categories. In practice, its use quickly becomes computationally prohibitive
because <code>repcnt</code> would need to be set very high for most applications.</p>
<p>CAUTION: "Hypergeometric" should NEVER be used for producing results for
biological interpretation.  If there is genuinely no bias in power to detect
DE in your experiment, the PWF will reflect this and the other methods will
produce accurate results.</p>
<p>"Hypergeometric" assumes there is no bias in power to detect differential
expression at all and calculates the p-values using a standard
hypergeometric distribution.  Useful if you wish to test the effect of
selection bias on your results.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Young, M. D., Wakefield, M. J., Smyth, G. K., Oshlack, A. (2010)
<em>Gene ontology analysis for RNA-seq: accounting for selection bias</em>
Genome Biology Date: Feb 2010 Vol: 11 Issue: 2 Pages: R14</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="nullp.html">nullp</a></code>, <code><a href="getgo.html">getgo</a></code>, <code><a href="getlength.html">getlength</a></code></p></div>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Matthew D. Young <a href="mailto:myoung@wehi.edu.au">myoung@wehi.edu.au</a></p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">genes</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">pwf</span> <span class="op">&lt;-</span> <span class="fu"><a href="nullp.html">nullp</a></span><span class="op">(</span><span class="va">genes</span>,<span class="st">'hg19'</span>,<span class="st">'ensGene'</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading hg19 length data...</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>initial point very close to some inequality constraints</span>
<span class="r-plt img"><img src="goseq-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="va">pvals</span> <span class="op">&lt;-</span> <span class="fu">goseq</span><span class="op">(</span><span class="va">pwf</span>,<span class="st">'hg19'</span>,<span class="st">'ensGene'</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Fetching GO annotations...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> For 9343 genes, we could not find any categories. These genes will be excluded.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> To force their use, please run with use_genes_without_cat=TRUE (see documentation).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> This was the default behavior for version 1.15.1 and earlier.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Calculating the p-values...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 'select()' returned 1:1 mapping between keys and columns</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pvals</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         category over_represented_pvalue under_represented_pvalue numDEInCat</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2490  GO:0005737            2.559737e-10                        1       2121</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 123   GO:0000278            2.132274e-08                        1        233</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9061  GO:0035556            3.612206e-08                        1        538</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3165  GO:0006793            1.365907e-07                        1        519</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12646 GO:0051301            5.097381e-07                        1        168</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17061 GO:0110165            6.456702e-07                        1       2773</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       numInCat                              term ontology</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2490      9352                         cytoplasm       CC</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 123        787                mitotic cell cycle       BP</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 9061      2029 intracellular signal transduction       BP</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3165      2013      phosphorus metabolic process       BP</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 12646      557                     cell division       BP</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 17061    12873        cellular anatomical entity       CC</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Matthew Young, Nadia Davidson, Federico Marini.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer></div>

  

  

  </body></html>

