<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Probability Weighting Function — nullp • goseq</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Probability Weighting Function — nullp"><meta name="description" content="Calculates a Probability Weighting Function for a set of genes based on a
given set of biased data (usually gene length) and each genes status as
differentially expressed or not."><meta property="og:description" content="Calculates a Probability Weighting Function for a set of genes based on a
given set of biased data (usually gene length) and each genes status as
differentially expressed or not."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">
    <a class="navbar-brand me-2" href="../index.html">goseq</a>
    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.57.2</small>
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>
    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/federicomarini/goseq/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <h1>Probability Weighting Function</h1>
      <small class="dont-index">Source: <a href="https://github.com/federicomarini/goseq/blob/devel/R/nullp.R" class="external-link"><code>R/nullp.R</code></a></small>
      <div class="d-none name"><code>nullp.Rd</code></div>
    </div>
    <div class="ref-description section level2">
    <p>Calculates a Probability Weighting Function for a set of genes based on a
given set of biased data (usually gene length) and each genes status as
differentially expressed or not.</p>
    </div>
    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">nullp</span><span class="op">(</span><span class="va">DEgenes</span>, <span class="va">genome</span>, <span class="va">id</span>, bias.data <span class="op">=</span> <span class="cn">NULL</span>, plot.fit <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
    </div>
    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
<dl><dt id="arg-degenes">DEgenes<a class="anchor" aria-label="anchor" href="#arg-degenes"></a></dt>
<dd><p>A named binary vector where 1 represents DE, 0 not DE and the
names are gene IDs.</p></dd>
<dt id="arg-genome">genome<a class="anchor" aria-label="anchor" href="#arg-genome"></a></dt>
<dd><p>A string identifying the genome that <code>genes</code> refer to.
For a list of supported organisms run <code>supportedGenomes</code>.</p></dd>
<dt id="arg-id">id<a class="anchor" aria-label="anchor" href="#arg-id"></a></dt>
<dd><p>A string identifying the gene identifier used by <code>genes</code>.
For a list of supported gene IDs run <code>supportedGeneIDs</code>.</p></dd>
<dt id="arg-bias-data">bias.data<a class="anchor" aria-label="anchor" href="#arg-bias-data"></a></dt>
<dd><p>A numeric vector containing the data on which the DE may
depend.  Usually this is the median transcript length of each gene in bp.
If set to <code>NULL</code> <code>nullp</code> will attempt to fetch length using
<code><a href="getlength.html">getlength</a></code>.</p></dd>
<dt id="arg-plot-fit">plot.fit<a class="anchor" aria-label="anchor" href="#arg-plot-fit"></a></dt>
<dd><p>Plot the PWF or not?  Calls <code><a href="plotPWF.html">plotPWF</a></code> with
default values if <code>TRUE</code>.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A data frame with 3 columns, named "DEgenes", "bias.data" and "pwf"
with the rownames set to the gene names.  Each row corresponds to a gene
with the DEgenes column specifying if the gene is DE (1 for DE, 0 for not
DE), the bias.data column giving the numeric value of the DE bias being
accounted for (usually the gene length or number of counts) and the pwf
column giving the genes value on the probability weighting function. This
object is usually passed to <code>goseq</code> to calculate enriched categories or
<code>plotPWF</code> for further plotting.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>It is essential that the entire analysis pipeline, from summarizing raw
reads through to using <code>goseq</code> be done in just one gene identifier
format.  If your data is in a different format you will need to obtain the
gene lengths and supply them to the <code>nullp</code> function using the
<code>bias.data</code> argument.  Converting to a supported format from another
format should be avoided whenever possible as this will almost always result
in data loss.</p>
<p><code>NA</code>s are allowed in the bias.data vector if you do not have
information about a certain gene.  Setting a gene to <code>NA</code> is preferable
to removing it from the analysis.</p>
<p>If <code>bias.data</code> is left as <code>NULL</code>, <code>nullp</code> attempts to use
<code><a href="getlength.html">getlength</a></code> to fetch GO category to gene identifier mappings.</p>
<p>It is recommended you review the fit produced by the <code>nullp</code> function
before proceeding by leaving <code>plot.fit</code> as <code>TRUE</code>.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Young, M. D., Wakefield, M. J., Smyth, G. K., Oshlack, A. (2010)
<em>Gene ontology analysis for RNA-seq: accounting for selection bias</em>
Genome Biology Date: Feb 2010 Vol: 11 Issue: 2 Pages: R14</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code>supportedGenomes</code>, <code>supportedGeneIDs</code>,
<code><a href="goseq.html">goseq</a></code>, <code><a href="getlength.html">getlength</a></code></p></div>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Matthew D. Young <a href="mailto:myoung@wehi.edu.au">myoung@wehi.edu.au</a></p>
    </div>
    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">genes</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">pwf</span> <span class="op">&lt;-</span> <span class="fu">nullp</span><span class="op">(</span><span class="va">genes</span>, <span class="st">'hg19'</span>, <span class="st">'ensGene'</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading hg19 length data...</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>initial point very close to some inequality constraints</span>
<span class="r-plt img"><img src="nullp-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>
    <footer><div class="pkgdown-footer-left">
  <p>Developed by Matthew Young, Nadia Davidson, Federico Marini.</p>
</div>
<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.9000.</p>
</div>
    </footer></div>
  </body></html>

