<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Plot the Probability Weighting Function — plotPWF • goseq</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Plot the Probability Weighting Function — plotPWF"><meta name="description" content="Plots the Probability Weighting Function created by nullp by
binning together genes."><meta property="og:description" content="Plots the Probability Weighting Function created by nullp by
binning together genes."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">
    <a class="navbar-brand me-2" href="../index.html">goseq</a>
    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.57.2</small>
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>
    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/federicomarini/goseq/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <h1>Plot the Probability Weighting Function</h1>
      <small class="dont-index">Source: <a href="https://github.com/federicomarini/goseq/blob/devel/R/plotPWF.R" class="external-link"><code>R/plotPWF.R</code></a></small>
      <div class="d-none name"><code>plotPWF.Rd</code></div>
    </div>
    <div class="ref-description section level2">
    <p>Plots the Probability Weighting Function created by <code><a href="nullp.html">nullp</a></code> by
binning together genes.</p>
    </div>
    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">plotPWF</span><span class="op">(</span></span>
<span>  <span class="va">pwf</span>,</span>
<span>  binsize <span class="op">=</span> <span class="st">"auto"</span>,</span>
<span>  pwf_col <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  pwf_lwd <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  xlab <span class="op">=</span> <span class="st">"Biased Data in &lt;binsize&gt; gene bins."</span>,</span>
<span>  ylab <span class="op">=</span> <span class="st">"Proportion DE"</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>
    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
<dl><dt id="arg-pwf">pwf<a class="anchor" aria-label="anchor" href="#arg-pwf"></a></dt>
<dd><p>A data frame with 3 columns named DEgenes, bias.data &amp; pwf and
row names giving the gene names.  Usually generated by <code><a href="nullp.html">nullp</a></code>.</p></dd>
<dt id="arg-binsize">binsize<a class="anchor" aria-label="anchor" href="#arg-binsize"></a></dt>
<dd><p>Calculate and plot the fraction of genes that are DE in bins
of this size.  If set to "auto" the best binsize for visualization is
attempted to be found automatically.</p></dd>
<dt id="arg-pwf-col">pwf_col<a class="anchor" aria-label="anchor" href="#arg-pwf-col"></a></dt>
<dd><p>The colour of the probability weighting function</p></dd>
<dt id="arg-pwf-lwd">pwf_lwd<a class="anchor" aria-label="anchor" href="#arg-pwf-lwd"></a></dt>
<dd><p>The width of the probability weighting function</p></dd>
<dt id="arg-xlab">xlab<a class="anchor" aria-label="anchor" href="#arg-xlab"></a></dt>
<dd><p>The x-axis label. <code>&lt;binsize&gt;</code> is replaced by the binsize
used.</p></dd>
<dt id="arg-ylab">ylab<a class="anchor" aria-label="anchor" href="#arg-ylab"></a></dt>
<dd><p>The y-axis label.</p></dd>
<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Extra arguments that are passed to plot.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>Nothing is returned.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This function is almost always called using the output from the
<code><a href="nullp.html">nullp</a></code> function.  However, it can be used to visualize the
length (or any other type of quantifiable) bias in ability to detect DE in a
data set.  The <code>pwf</code> argument needs to be a data frame with 3 columns
each containing numeric entries (although NAs are permitted in the bias.data
and pwf columns), which must be named "DEgenes", "bias.data" and "pwf",
although they can appear in any order.  The row names are taken to be the
gene names. The DEgenes column should be 0s or 1s where 1 represents a DE
gene, 0 a gene which is not DE.  The bias.data column is a quantification of
the quantity for which there is a bias in detecting DE for the associated
gene, this is usually gene length or the number of counts associated with a
gene.  Finally, the pwf column gives the probability weighting to be applied
for a given gene.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Young, M. D., Wakefield, M. J., Smyth, G. K., Oshlack, A. (2010)
<em>Gene ontology analysis for RNA-seq: accounting for selection bias</em>
Genome Biology Date: Feb 2010 Vol: 11 Issue: 2 Pages: R14</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="nullp.html">nullp</a></code></p></div>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Matthew D. Young <a href="mailto:myoung@wehi.edu.au">myoung@wehi.edu.au</a></p>
    </div>
    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">genes</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">pwf</span> <span class="op">&lt;-</span> <span class="fu"><a href="nullp.html">nullp</a></span><span class="op">(</span><span class="va">genes</span>, <span class="st">'hg19'</span>, <span class="st">'ensGene'</span>,plot.fit<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading hg19 length data...</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>initial point very close to some inequality constraints</span>
<span class="r-in"><span><span class="fu">plotPWF</span><span class="op">(</span><span class="va">pwf</span>,binsize<span class="op">=</span><span class="fl">200</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="plotPWF-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>
    <footer><div class="pkgdown-footer-left">
  <p>Developed by Matthew Young, Nadia Davidson, Federico Marini.</p>
</div>
<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.9000.</p>
</div>
    </footer></div>
  </body></html>

