<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="goseq">
<title>`goseq`: Gene Ontology testing for RNA-seq datasets
 • goseq</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="`goseq`: Gene Ontology testing for RNA-seq datasets
">
<meta property="og:description" content="goseq">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">goseq</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.57.3</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/goseq_manual.html">`goseq`: Gene Ontology testing for RNA-seq datasets
</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/federicomarini/goseq/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>`goseq`: Gene Ontology testing for RNA-seq datasets
</h1>
                        <h4 data-toc-skip class="author">Matthew D.
Young</h4>
                                    <h4 data-toc-skip class="author">Nadia
Davidson</h4>
                                    <h4 data-toc-skip class="author">Matthew
J. Wakefield</h4>
                                    <h4 data-toc-skip class="author">Gordon
K. Smyth</h4>
                                    <h4 data-toc-skip class="author">Alicia
Oshlack</h4>
                        
            <h4 data-toc-skip class="date">12 June 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/federicomarini/goseq/blob/HEAD/vignettes/goseq_manual.Rmd" class="external-link"><code>vignettes/goseq_manual.Rmd</code></a></small>
      <div class="d-none name"><code>goseq_manual.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This document gives an introduction to the use of the <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em> R
Bioconductor package <span class="citation">(Young et al. 2010)</span>.
This package provides methods for performing Gene Ontology analysis of
RNA-seq data, taking length bias into account <span class="citation">(Oshlack and Wakefield 2009)</span>. The methods and
software used by <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em> are
equally applicable to other category based test of RNA-seq data, such as
KEGG pathway analysis.</p>
<p>To install <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em>, you
can run the following commands:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"goseq"</span><span class="op">)</span></span></code></pre></div>
<p>If you want to install the development version from GitHub, you can
alternatively run this command:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"federicomarini/goseq"</span><span class="op">)</span></span></code></pre></div>
<p>TODO: name sections and crossreference through them</p>
<p>Once installed, the <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em>
package can be easily loaded into R using:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/federicomarini/goseq" class="external-link">"goseq"</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: BiasedUrn</span></span>
<span><span class="co">#&gt; Loading required package: geneLenDataBase</span></span>
<span><span class="co">#&gt; </span></span></code></pre></div>
<p>In order to perform a GO analysis of your RNA-seq data, <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em> only
requires a simple named vector, which contains two pieces of
information.</p>
<ul>
<li>
<code>Measured genes</code>: all genes for which RNA-seq data was
gathered for your experiment. Each element of your vector should be
named by a unique gene identifier.</li>
<li>
<code>Differentially expressed genes</code>: each element of your
vector should be either a 1 or a 0, where 1 indicates that the gene is
differentially expressed and 0 that it is not.</li>
</ul>
<p>If the organism, gene identifier or category test is currently not
natively supported by <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em>, it
will also be necessary to supply additional information regarding the
genes length and/or the association between categories and genes.</p>
<p>Bioconductor R packages such as <em><a href="https://bioconductor.org/packages/3.19/Rsubread" class="external-link">Rsubread</a></em>
allow for the summarization of mapped reads into a table of counts, such
as reads per gene. From there, several packages exist for performing
differential expression analysis on summarized data (e.g. <em><a href="https://bioconductor.org/packages/3.19/edgeR" class="external-link">edgeR</a></em>,
<span class="citation">(Robinson and Smyth 2007, 2008; Robinson,
McCarthy, and Smyth 2010)</span>).</p>
<p><em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em> will
work with any method for determining differential expression and as such
differential expression analysis is outside the scope of this document,
but in order to facilitate ease of use, we will make use of the <em><a href="https://bioconductor.org/packages/3.19/edgeR" class="external-link">edgeR</a></em>
package to calculate differentially expressed (DE) genes in all the case
studies in this document.</p>
</div>
<div class="section level2">
<h2 id="reading-data">Reading data<a class="anchor" aria-label="anchor" href="#reading-data"></a>
</h2>
<p>We assume that the user can use appropriate in-built <code>R</code>
functions (such as <code>read.table</code> or <code>scan</code>) to
obtain two vectors, one containing all genes assayed in the RNA-seq
experiment, the other containing all genes which are DE. If we assume
that the vector of genes being assayed is named
<code>assayed.genes</code> and the vector of DE genes is named
<code>de.genes</code> we can construct a named vector suitable for use
with <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em> using
the following:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene.vector</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html" class="external-link">as.integer</a></span><span class="op">(</span><span class="va">assayed.genes</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">de.genes</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gene.vector</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">assayed.genes</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene.vector</span><span class="op">)</span></span></code></pre></div>
<p>It may be that the user can already read in a vector in this format,
in which case it can then be immediately used by <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em>.</p>
</div>
<div class="section level2">
<h2 id="go-testing-of-rna-seq-data">GO testing of RNA-seq data<a class="anchor" aria-label="anchor" href="#go-testing-of-rna-seq-data"></a>
</h2>
<p>To begin the analysis, <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em> first
needs to quantify the length bias present in the dataset under
consideration. This is done by calculating a Probability Weighting
Function or PWF which can be thought of as a function which gives the
probability that a gene will be differentially expressed (DE), based on
its length alone. The PWF is calculated by fitting a monotonic spline to
the binary data series of differential expression (1=DE, 0=Not DE) as a
function of gene length. The PWF is used to weight the chance of
selecting each gene when forming a null distribution for GO category
membership. The fact that the PWF is calculated directly from the
dataset under consideration makes this approach robust, only correcting
for the length bias present in the data. For example, if <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em> is
run on a microarray dataset, for which no length bias exists, the
calculated PWF will be nearly flat and all genes will be weighted
equally, resulting in no length bias correction.</p>
<p>In order to account for the length bias inherent to RNA-seq data when
performing a GO analysis (or other category based tests), one cannot
simply use the hypergeometric distribution as the null distribution for
category membership, which is appropriate for data without DE length
bias, such as microarray data. GO analysis of RNA-seq data requires the
use of random sampling in order to generate a suitable null distribution
for GO category membership and calculate each categories significance
for over representation amongst DE genes.</p>
<p>However, this random sampling is computationally expensive. In most
cases, the Wallenius distribution can be used to approximate the true
null distribution, without any significant loss in accuracy. The <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em>
package implements this approximation as its default option. The option
to generate the null distribution using random sampling is also included
as an option, but users should be aware that the default number of
samples generated will not be enough to accurately call enrichment when
there are a large number of go terms.</p>
<p>Having established a null distribution, each GO category is then
tested for over and under representation amongst the set of
differentially expressed genes and the null is used to calculate a
p-value for under and over representation.</p>
</div>
<div class="section level2">
<h2 id="natively-supported-gene-identifiers-and-category-tests">Natively supported Gene Identifiers and category tests<a class="anchor" aria-label="anchor" href="#natively-supported-gene-identifiers-and-category-tests"></a>
</h2>
<p><em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em> needs
to know the length of each gene, as well as what GO categories (or other
categories of interest) each gene is associated with. <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em>
relies on the UCSC genome browser to provide the length information for
each gene. However, because the process of fetching the length of every
transcript is slow and bandwidth intensive, <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em>
relies on an offline copy of this information stored in the data package
<em><a href="https://bioconductor.org/packages/3.19/geneLenDataBase" class="external-link">geneLenDataBase</a></em>.
To see which genome/gene identifier combinations are in the local
database, simply run:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/supportedOrganisms.html">supportedOrganisms</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>The leftmost columns in the output of this command list the genomes
and gene identifiers respectively. If length data exists in the local
database it is indicated in the second last column. If your genome/ID
combination is not in the local database, it may be downloaded from the
UCSC genome browser or taken from a <code>TxDb</code> annotation package
(if installed). If your genome/ID combination is not found in any
database, you will have to manually specify the gene lengths. We
encourage all users to manually specify their gene lengths if provided
by upstream summarization programs. e.g. <code>featureCounts</code>, as
these lengths will be more accurate.</p>
<p>In order to link GO categories to genes, <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em> uses
the organism packages from Bioconductor. These packages are named
<code>org.&lt;Genome&gt;.&lt;ID&gt;.db</code>, where
<code>&lt;Genome&gt;</code> is a short string identifying the genome and
<code>&lt;ID&gt;</code> is a short string identifying the gene
identifier.<br>
Currently, <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em> will
automatically retrieve the mapping between GO categories and genes from
the relevant package (as long as it is installed) for commonly used
genome/ID combinations. If GO mappings are not automatically available
for your genome/ID combination, you will have to manually specify the
relationship between genes and categories. Although the Genome/ID naming
conventions used by the organism packages differ from the UCSC, <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em> is
able to convert between the two, so the user need only ever specify the
UCSC genome/ID in most cases. The final column indicates whether the
Genome/ID combination is supported for GO categories.</p>
</div>
<div class="section level2">
<h2 id="non-native-gene-identifier-or-category-test">Non-native Gene Identifier or category test<a class="anchor" aria-label="anchor" href="#non-native-gene-identifier-or-category-test"></a>
</h2>
<p>If the organism, Gene Identifier or category test you wish to perform
is not in the native <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em>
database, you will have to supply one or all of the following:</p>
<ul>
<li>
<code>Length data</code>: the length of each gene in your gene
identifier format.</li>
<li>
<code>Category mappings</code>: the mapping (usually many-to-many)
between the categories you wish to test for over/under representation
amongst DE genes and genes in your gene identifier format.</li>
</ul>
<div class="section level3">
<h3 id="length-data-format">Length data format<a class="anchor" aria-label="anchor" href="#length-data-format"></a>
</h3>
<p>The length data must be formatted as a numeric vector, of the same
length as the main named vector specifying gene names/DE genes. Each
entry should give the length of the corresponding gene in bp. If length
data is unavailable for some genes, that entry should be set to
<code>NA</code>.</p>
</div>
<div class="section level3">
<h3 id="category-mapping-format">Category mapping format<a class="anchor" aria-label="anchor" href="#category-mapping-format"></a>
</h3>
<p>The mapping between category names and genes should be given as a
data frame with two columns. One column should contain the gene IDs and
the other the name of an associated category. As the mapping between
categories and genes is usually many-to-many, this data frame will
usually have multiple rows with the same gene name and category
name.</p>
<p>Alternatively, mappings between genes and categories can be given as
a list. The names of list entries should be gene IDs and the entries
themselves should be a vector of category names to which the gene ID
corresponds.</p>
</div>
<div class="section level3">
<h3 id="some-additional-tips">Some additional tips<a class="anchor" aria-label="anchor" href="#some-additional-tips"></a>
</h3>
<p>Any organism for which there is an annotation on either Ensembl or
the UCSC, can be easily turned into length data using the <em><a href="https://bioconductor.org/packages/3.19/txdbmaker" class="external-link">txdbmaker</a></em>
package. To do this, first create a <code>TranscriptDb</code> object
using either <code>makeTxDbFromBiomart</code> or
<code>makeTxDbFromUCSC</code> (see the help in the <em><a href="https://bioconductor.org/packages/3.19/txdbmaker" class="external-link">txdbmaker</a></em>
package on using these commands). Once you have a
<code>TranscriptDb</code> object, you can get a vector named by gene ID
containing the median transcript length of each gene simply by using the
command.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">txsByGene</span> <span class="op">&lt;-</span> <span class="fu">transcriptsBy</span><span class="op">(</span><span class="va">txdb</span>, <span class="st">"gene"</span><span class="op">)</span></span>
<span><span class="va">lengthData</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/median.html" class="external-link">median</a></span><span class="op">(</span><span class="fu">width</span><span class="op">(</span><span class="va">txsByGene</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>The relationship between gene identifier and GO category can usually
be obtained from the Gene Ontology website (www.geneontology.org) or
from the NCBI. Additionally, the Bioconductor <em><a href="https://bioconductor.org/packages/3.19/AnnotationDbi" class="external-link">AnnotationDbi</a></em>
library has recently added a function
<code>makeOrgPackageFromNCBI</code>, which can be used to create an
organism package from within R, using the NCBI data. TODO: is this still
current? Once created, this package can then be used to obtain the
mapping between genes and Gene Ontology.</p>
</div>
</div>
<div class="section level2">
<h2 id="case-study-prostate-cancer-data">Case study: Prostate cancer data<a class="anchor" aria-label="anchor" href="#case-study-prostate-cancer-data"></a>
</h2>
<div class="section level3">
<h3 id="introduction-1">Introduction<a class="anchor" aria-label="anchor" href="#introduction-1"></a>
</h3>
<p>This section provides an analysis of data from an RNA-seq experiment
to illustrate the use of <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em> for
GO analysis.</p>
<p>This experiment examined the effects of androgen stimulation on a
human prostate cancer cell line, LNCaP. The data set includes more than
17 million short cDNA reads obtained for both the treated and untreated
cell line and sequenced on Illumina’s 1G genome analyzer.</p>
<p>For each sample we were provided with the raw 35 bp RNA-seq reads
from the authors. For the untreated prostate cancer cells (LNCaP cell
line) there were 4 lanes totaling ~10 million, 35 bp reads. For the
treated cells there were 3 lanes totaling ~7 million, 35 bp reads. All
replicates were technical replicates. Reads were mapped to NCBI version
36.3 of the human genome using <code>bowtie</code>.<br>
Any read with which mapped to multiple locations was discarded. Using
the ENSEMBL 54 annotation from biomart, each mapped read was associated
with an ENSEMBL gene. This was done by associating any read that
overlapped with any part of the gene (not just the exons) with that
gene. Reads that did not correspond to genes were discarded.</p>
</div>
<div class="section level3">
<h3 id="source-of-the-data">Source of the data<a class="anchor" aria-label="anchor" href="#source-of-the-data"></a>
</h3>
<p>The data set used in this case study is taken from <span class="citation">(Li et al. 2008)</span> and was made available from the
authors upon request.</p>
</div>
<div class="section level3">
<h3 id="determining-the-de-genes-using-edger">Determining the DE genes using <em><a href="https://bioconductor.org/packages/3.19/edgeR" class="external-link">edgeR</a></em><a class="anchor" aria-label="anchor" href="#determining-the-de-genes-using-edger"></a>
</h3>
<p>To begin with, we load in the text data and convert it the
appropriate <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em>
<code>DGEList</code> object.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioinf.wehi.edu.au/edgeR/" class="external-link">"edgeR"</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: limma</span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">table.summary</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"Li_sum.txt"</span>, package <span class="op">=</span> <span class="st">"goseq"</span><span class="op">)</span>,</span>
<span>  sep <span class="op">=</span> <span class="st">"\t"</span>, header <span class="op">=</span> <span class="cn">TRUE</span>, stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="va">counts</span> <span class="op">&lt;-</span> <span class="va">table.summary</span><span class="op">[</span>, <span class="op">-</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">table.summary</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span></span>
<span><span class="va">grp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Control"</span>, <span class="st">"Treated"</span><span class="op">)</span>, times <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">summarized</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html" class="external-link">DGEList</a></span><span class="op">(</span><span class="va">counts</span>, lib.size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">colSums</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span>, group <span class="op">=</span> <span class="va">grp</span><span class="op">)</span></span></code></pre></div>
<p>Next, we use <em><a href="https://bioconductor.org/packages/3.19/edgeR" class="external-link">edgeR</a></em> to
estimate the biological dispersion and calculate differential expression
using a negative binomial model.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">disp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/estimateCommonDisp.html" class="external-link">estimateCommonDisp</a></span><span class="op">(</span><span class="va">summarized</span><span class="op">)</span></span>
<span><span class="va">disp</span><span class="op">$</span><span class="va">common.dispersion</span></span>
<span><span class="co">#&gt; [1] 0.05688364</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tested</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/exactTest.html" class="external-link">exactTest</a></span><span class="op">(</span><span class="va">disp</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/topTags.html" class="external-link">topTags</a></span><span class="op">(</span><span class="va">tested</span><span class="op">)</span></span>
<span><span class="co">#&gt; Comparison of groups:  Treated-Control </span></span>
<span><span class="co">#&gt;                     logFC   logCPM       PValue          FDR</span></span>
<span><span class="co">#&gt; ENSG00000127954 11.557868 6.680748 2.574972e-80 1.274766e-75</span></span>
<span><span class="co">#&gt; ENSG00000151503  5.398963 8.499530 1.781732e-65 4.410321e-61</span></span>
<span><span class="co">#&gt; ENSG00000096060  4.897600 9.446705 7.983755e-60 1.317479e-55</span></span>
<span><span class="co">#&gt; ENSG00000091879  5.737627 6.282646 1.207655e-54 1.494654e-50</span></span>
<span><span class="co">#&gt; ENSG00000132437 -5.880436 7.951910 2.950042e-52 2.920896e-48</span></span>
<span><span class="co">#&gt; ENSG00000166451  4.564246 8.458467 7.126762e-52 5.880291e-48</span></span>
<span><span class="co">#&gt; ENSG00000131016  5.254737 6.607957 1.066807e-51 7.544765e-48</span></span>
<span><span class="co">#&gt; ENSG00000163492  7.085400 5.128514 2.716461e-45 1.681014e-41</span></span>
<span><span class="co">#&gt; ENSG00000113594  4.051053 8.603264 9.272066e-44 5.100254e-40</span></span>
<span><span class="co">#&gt; ENSG00000116285  4.108522 7.864773 6.422468e-43 3.179507e-39</span></span></code></pre></div>
<p>Finally, we format the DE genes into a vector suitable for use with
<em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/integer.html" class="external-link">as.integer</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span><span class="va">tested</span><span class="op">$</span><span class="va">table</span><span class="op">$</span><span class="va">PValue</span><span class="op">[</span><span class="va">tested</span><span class="op">$</span><span class="va">table</span><span class="op">$</span><span class="va">logFC</span> <span class="op">!=</span> <span class="fl">0</span><span class="op">]</span>,</span>
<span>           method <span class="op">=</span> <span class="st">"BH"</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">.05</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">genes</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">tested</span><span class="op">$</span><span class="va">table</span><span class="op">[</span><span class="va">tested</span><span class="op">$</span><span class="va">table</span><span class="op">$</span><span class="va">logFC</span> <span class="op">!=</span> <span class="fl">0</span>, <span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">genes</span><span class="op">)</span></span>
<span><span class="co">#&gt; genes</span></span>
<span><span class="co">#&gt;     0     1 </span></span>
<span><span class="co">#&gt; 19535  3208</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="determining-genome-and-gene-id">Determining Genome and Gene ID<a class="anchor" aria-label="anchor" href="#determining-genome-and-gene-id"></a>
</h3>
<p>In order to allow for automatic data retrieval, the user has to tell
<em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em> what
genome and gene ID format were used to summarize the data. In our case
we will use the <code>hg19</code> build of the human genome, we check
what code this corresponds to by running:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/supportedOrganisms.html">supportedOrganisms</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;          Genome         Id  Id Description Lengths in geneLeneDataBase</span></span>
<span><span class="co">#&gt; 1       anoCar1    ensGene Ensembl gene ID                        TRUE</span></span>
<span><span class="co">#&gt; 2       anoGam1    ensGene Ensembl gene ID                        TRUE</span></span>
<span><span class="co">#&gt; 98      anoGam3                                                  FALSE</span></span>
<span><span class="co">#&gt; 3       apiMel2    ensGene Ensembl gene ID                        TRUE</span></span>
<span><span class="co">#&gt; 138 Arabidopsis                                                  FALSE</span></span>
<span><span class="co">#&gt; 56      bosTau2 geneSymbol     Gene Symbol                        TRUE</span></span>
<span><span class="co">#&gt;     GO Annotation Available</span></span>
<span><span class="co">#&gt; 1                     FALSE</span></span>
<span><span class="co">#&gt; 2                      TRUE</span></span>
<span><span class="co">#&gt; 98                     TRUE</span></span>
<span><span class="co">#&gt; 3                     FALSE</span></span>
<span><span class="co">#&gt; 138                    TRUE</span></span>
<span><span class="co">#&gt; 56                     TRUE</span></span></code></pre></div>
<p>Which lists the genome codes in the far left column, headed
<code>Genome</code>. As we are using <code>hg19</code> and we also know
that we used ENSEMBL Gene ID to summarize our read data, we check what
code this corresponds to by running:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/supportedOrganisms.html">supportedOrganisms</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="../reference/supportedOrganisms.html">supportedOrganisms</a></span><span class="op">(</span><span class="op">)</span><span class="op">$</span><span class="va">Genome</span> <span class="op">==</span> <span class="st">"hg19"</span>, <span class="op">]</span></span>
<span><span class="co">#&gt;    Genome         Id  Id Description Lengths in geneLeneDataBase</span></span>
<span><span class="co">#&gt; 25   hg19    ensGene Ensembl gene ID                        TRUE</span></span>
<span><span class="co">#&gt; 48   hg19  knownGene  Entrez Gene ID                        TRUE</span></span>
<span><span class="co">#&gt; 81   hg19 geneSymbol     Gene Symbol                        TRUE</span></span>
<span><span class="co">#&gt;    GO Annotation Available</span></span>
<span><span class="co">#&gt; 25                    TRUE</span></span>
<span><span class="co">#&gt; 48                    TRUE</span></span>
<span><span class="co">#&gt; 81                    TRUE</span></span></code></pre></div>
<p>The gene ID codes are listed in the column second from left, titled
<code>Id</code>. We find that our gene ID code is <code>ensGene</code>.
We will use these strings whenever we are asked for a genome or id. If
the gene ID is missing for your Genome (for example this is the case for
<code>hg38</code>), then the genome is not supported in the <em><a href="https://bioconductor.org/packages/3.19/geneLenDataBase" class="external-link">geneLenDataBase</a></em>
package. Gene lengths will either be automatically fetched from TxDB,
UCSC or you will need to provide them manually. Supported Gene IDs to
automatically fetch GO terms should usually either be Entrez
(<code>knownGene</code>), Ensembl (<code>ensGene</code>) or gene symbols
(<code>geneSymbol</code>).</p>
</div>
<div class="section level3">
<h3 id="go-analysis">GO analysis<a class="anchor" aria-label="anchor" href="#go-analysis"></a>
</h3>
<div class="section level4">
<h4 id="fitting-the-probability-weighting-function-pwf">Fitting the Probability Weighting Function (PWF)<a class="anchor" aria-label="anchor" href="#fitting-the-probability-weighting-function-pwf"></a>
</h4>
<p>We first need to obtain a weighting for each gene, depending on its
length, given by the PWF. As you may have noticed when running
<code><a href="https://rdrr.io/pkg/geneLenDataBase/man/supportedGenomes.html" class="external-link">supportedGenomes()</a></code> or <code><a href="https://rdrr.io/pkg/geneLenDataBase/man/supportedGeneIDs.html" class="external-link">supportedGeneIDs()</a></code>,
length data is available in the local database for our gene ID,
<code>ensGene</code>, and our genome, <code>hg19</code>. We will let
<em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em>
automatically fetch this data from its databases.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pwf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/nullp.html">nullp</a></span><span class="op">(</span><span class="va">genes</span>, <span class="st">"hg19"</span>, <span class="st">"ensGene"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading hg19 length data...</span></span>
<span><span class="co">#&gt; Warning in pcls(G): initial point very close to some inequality constraints</span></span></code></pre></div>
<p><img src="goseq_manual_files/figure-html/pwf-1.png" width="768"></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pwf</span><span class="op">)</span></span>
<span><span class="co">#&gt;                 DEgenes bias.data        pwf</span></span>
<span><span class="co">#&gt; ENSG00000230758       0       247 0.03757470</span></span>
<span><span class="co">#&gt; ENSG00000182463       0      3133 0.20436865</span></span>
<span><span class="co">#&gt; ENSG00000124208       0      1978 0.16881769</span></span>
<span><span class="co">#&gt; ENSG00000230753       0       466 0.06927243</span></span>
<span><span class="co">#&gt; ENSG00000224628       0      1510 0.15903532</span></span>
<span><span class="co">#&gt; ENSG00000125835       0       954 0.12711992</span></span></code></pre></div>
<p><code><a href="../reference/nullp.html">nullp()</a></code> plots the resulting fit, allowing verification
of the goodness of fit before continuing the analysis. Further plotting
of the pwf can be performed using the <code><a href="../reference/plotPWF.html">plotPWF()</a></code>
function.</p>
<p>The output of <code><a href="../reference/nullp.html">nullp()</a></code> contains all the data used to
create the PWF, as well as the PWF itself. It is a data frame with 3
columns, named <code>DEgenes</code>, <code>bias.data</code> and
<code>pwf</code> with the rownames set to the gene names. Each row
corresponds to a gene with the <code>DEgenes</code> column specifying if
the gene is DE (1 for DE, 0 for not DE), the <code>bias.data</code>
column giving the numeric value of the DE bias being accounted for
(usually the gene length or number of counts), and the <code>pwf</code>
column giving the genes value on the probability weighting function.</p>
</div>
<div class="section level4">
<h4 id="using-the-wallenius-approximation">Using the Wallenius approximation<a class="anchor" aria-label="anchor" href="#using-the-wallenius-approximation"></a>
</h4>
<p>To start with we will use the default method, to calculate the over
and under expressed GO categories among DE genes. Again, we allow <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em> to
fetch data automatically, except this time the data being fetched is the
relationship between ENSEMBL gene IDs and GO categories.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">GO.wall</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/goseq.html">goseq</a></span><span class="op">(</span><span class="va">pwf</span>, <span class="st">"hg19"</span>, <span class="st">"ensGene"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Fetching GO annotations...</span></span>
<span><span class="co">#&gt; Loading required package: AnnotationDbi</span></span>
<span><span class="co">#&gt; Loading required package: stats4</span></span>
<span><span class="co">#&gt; Loading required package: BiocGenerics</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'BiocGenerics'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:limma':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     plotMA</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">#&gt;     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">#&gt;     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply, match,</span></span>
<span><span class="co">#&gt;     mget, order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span></span>
<span><span class="co">#&gt;     rbind, Reduce, rownames, sapply, setdiff, table, tapply, union,</span></span>
<span><span class="co">#&gt;     unique, unsplit, which.max, which.min</span></span>
<span><span class="co">#&gt; Loading required package: Biobase</span></span>
<span><span class="co">#&gt; Welcome to Bioconductor</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Vignettes contain introductory material; view with</span></span>
<span><span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see 'citation("Biobase")',</span></span>
<span><span class="co">#&gt;     and for packages 'citation("pkgname")'.</span></span>
<span><span class="co">#&gt; Loading required package: IRanges</span></span>
<span><span class="co">#&gt; Loading required package: S4Vectors</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'S4Vectors'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:geneLenDataBase':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     unfactor</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:utils':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     findMatches</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     expand.grid, I, unname</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; For 9343 genes, we could not find any categories. These genes will be excluded.</span></span>
<span><span class="co">#&gt; To force their use, please run with use_genes_without_cat=TRUE (see documentation).</span></span>
<span><span class="co">#&gt; This was the default behavior for version 1.15.1 and earlier.</span></span>
<span><span class="co">#&gt; Calculating the p-values...</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">GO.wall</span><span class="op">)</span></span>
<span><span class="co">#&gt;         category over_represented_pvalue under_represented_pvalue numDEInCat</span></span>
<span><span class="co">#&gt; 2490  GO:0005737            1.898519e-10                        1       2025</span></span>
<span><span class="co">#&gt; 123   GO:0000278            5.486125e-09                        1        227</span></span>
<span><span class="co">#&gt; 2436  GO:0005615            5.196258e-08                        1        498</span></span>
<span><span class="co">#&gt; 17061 GO:0110165            6.638828e-08                        1       2646</span></span>
<span><span class="co">#&gt; 3165  GO:0006793            1.099583e-07                        1        498</span></span>
<span><span class="co">#&gt; 9061  GO:0035556            1.158245e-07                        1        512</span></span>
<span><span class="co">#&gt;       numInCat                              term ontology</span></span>
<span><span class="co">#&gt; 2490      9352                         cytoplasm       CC</span></span>
<span><span class="co">#&gt; 123        787                mitotic cell cycle       BP</span></span>
<span><span class="co">#&gt; 2436      2079               extracellular space       CC</span></span>
<span><span class="co">#&gt; 17061    12873        cellular anatomical entity       CC</span></span>
<span><span class="co">#&gt; 3165      2013      phosphorus metabolic process       BP</span></span>
<span><span class="co">#&gt; 9061      2029 intracellular signal transduction       BP</span></span></code></pre></div>
<p>The resulting object is ordered by GO category over representation
amongst DE genes.</p>
</div>
<div class="section level4">
<h4 id="using-random-sampling">Using random sampling<a class="anchor" aria-label="anchor" href="#using-random-sampling"></a>
</h4>
<p>It may sometimes be desirable to use random sampling to generate the
null distribution for category membership. For example, to check
consistency against results from the Wallenius approximation. This is
easily accomplished by using the <code>method</code> option to specify
sampling and the <code>repcnt</code> option to specify the number of
samples to generate:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">GO.samp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/goseq.html">goseq</a></span><span class="op">(</span><span class="va">pwf</span>, <span class="st">"hg19"</span>, <span class="st">"ensGene"</span>, method <span class="op">=</span> <span class="st">"Sampling"</span>, repcnt <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span>
<span><span class="co">#&gt; Fetching GO annotations...</span></span>
<span><span class="co">#&gt; For 9343 genes, we could not find any categories. These genes will be excluded.</span></span>
<span><span class="co">#&gt; To force their use, please run with use_genes_without_cat=TRUE (see documentation).</span></span>
<span><span class="co">#&gt; This was the default behavior for version 1.15.1 and earlier.</span></span>
<span><span class="co">#&gt; Running the simulation...</span></span>
<span><span class="co">#&gt; 0%   </span></span>
<span><span class="co">#&gt; 0%   </span></span>
<span><span class="co">#&gt; 0%   </span></span>
<span><span class="co">#&gt; 0%   </span></span>
<span><span class="co">#&gt; 0%   </span></span>
<span><span class="co">#&gt; 1%   </span></span>
<span><span class="co">#&gt; 1%   </span></span>
<span><span class="co">#&gt; 1%   </span></span>
<span><span class="co">#&gt; 1%   </span></span>
<span><span class="co">#&gt; 1%   </span></span>
<span><span class="co">#&gt; 1%   </span></span>
<span><span class="co">#&gt; 1%   </span></span>
<span><span class="co">#&gt; 1%   </span></span>
<span><span class="co">#&gt; 1%   </span></span>
<span><span class="co">#&gt; 2%   </span></span>
<span><span class="co">#&gt; 2%   </span></span>
<span><span class="co">#&gt; 2%   </span></span>
<span><span class="co">#&gt; 2%   </span></span>
<span><span class="co">#&gt; 2%   </span></span>
<span><span class="co">#&gt; 2%   </span></span>
<span><span class="co">#&gt; 2%   </span></span>
<span><span class="co">#&gt; 2%   </span></span>
<span><span class="co">#&gt; 2%   </span></span>
<span><span class="co">#&gt; 2%   </span></span>
<span><span class="co">#&gt; 2%   </span></span>
<span><span class="co">#&gt; 3%   </span></span>
<span><span class="co">#&gt; 3%   </span></span>
<span><span class="co">#&gt; 3%   </span></span>
<span><span class="co">#&gt; 3%   </span></span>
<span><span class="co">#&gt; 3%   </span></span>
<span><span class="co">#&gt; 3%   </span></span>
<span><span class="co">#&gt; 3%   </span></span>
<span><span class="co">#&gt; 3%   </span></span>
<span><span class="co">#&gt; 3%   </span></span>
<span><span class="co">#&gt; 4%   </span></span>
<span><span class="co">#&gt; 4%   </span></span>
<span><span class="co">#&gt; 4%   </span></span>
<span><span class="co">#&gt; 4%   </span></span>
<span><span class="co">#&gt; 4%   </span></span>
<span><span class="co">#&gt; 4%   </span></span>
<span><span class="co">#&gt; 4%   </span></span>
<span><span class="co">#&gt; 4%   </span></span>
<span><span class="co">#&gt; 4%   </span></span>
<span><span class="co">#&gt; 4%   </span></span>
<span><span class="co">#&gt; 4%   </span></span>
<span><span class="co">#&gt; 5%   </span></span>
<span><span class="co">#&gt; 5%   </span></span>
<span><span class="co">#&gt; 5%   </span></span>
<span><span class="co">#&gt; 5%   </span></span>
<span><span class="co">#&gt; 5%   </span></span>
<span><span class="co">#&gt; 5%   </span></span>
<span><span class="co">#&gt; 5%   </span></span>
<span><span class="co">#&gt; 5%   </span></span>
<span><span class="co">#&gt; 5%   </span></span>
<span><span class="co">#&gt; 6%   </span></span>
<span><span class="co">#&gt; 6%   </span></span>
<span><span class="co">#&gt; 6%   </span></span>
<span><span class="co">#&gt; 6%   </span></span>
<span><span class="co">#&gt; 6%   </span></span>
<span><span class="co">#&gt; 6%   </span></span>
<span><span class="co">#&gt; 6%   </span></span>
<span><span class="co">#&gt; 6%   </span></span>
<span><span class="co">#&gt; 6%   </span></span>
<span><span class="co">#&gt; 6%   </span></span>
<span><span class="co">#&gt; 6%   </span></span>
<span><span class="co">#&gt; 7%   </span></span>
<span><span class="co">#&gt; 7%   </span></span>
<span><span class="co">#&gt; 7%   </span></span>
<span><span class="co">#&gt; 7%   </span></span>
<span><span class="co">#&gt; 7%   </span></span>
<span><span class="co">#&gt; 7%   </span></span>
<span><span class="co">#&gt; 7%   </span></span>
<span><span class="co">#&gt; 7%   </span></span>
<span><span class="co">#&gt; 7%   </span></span>
<span><span class="co">#&gt; 8%   </span></span>
<span><span class="co">#&gt; 8%   </span></span>
<span><span class="co">#&gt; 8%   </span></span>
<span><span class="co">#&gt; 8%   </span></span>
<span><span class="co">#&gt; 8%   </span></span>
<span><span class="co">#&gt; 8%   </span></span>
<span><span class="co">#&gt; 8%   </span></span>
<span><span class="co">#&gt; 8%   </span></span>
<span><span class="co">#&gt; 8%   </span></span>
<span><span class="co">#&gt; 8%   </span></span>
<span><span class="co">#&gt; 8%   </span></span>
<span><span class="co">#&gt; 9%   </span></span>
<span><span class="co">#&gt; 9%   </span></span>
<span><span class="co">#&gt; 9%   </span></span>
<span><span class="co">#&gt; 9%   </span></span>
<span><span class="co">#&gt; 9%   </span></span>
<span><span class="co">#&gt; 9%   </span></span>
<span><span class="co">#&gt; 9%   </span></span>
<span><span class="co">#&gt; 9%   </span></span>
<span><span class="co">#&gt; 9%   </span></span>
<span><span class="co">#&gt; 10%   </span></span>
<span><span class="co">#&gt; 10%   </span></span>
<span><span class="co">#&gt; 10%   </span></span>
<span><span class="co">#&gt; 10%   </span></span>
<span><span class="co">#&gt; 10%   </span></span>
<span><span class="co">#&gt; 10%   </span></span>
<span><span class="co">#&gt; 10%   </span></span>
<span><span class="co">#&gt; 10%   </span></span>
<span><span class="co">#&gt; 10%   </span></span>
<span><span class="co">#&gt; 10%   </span></span>
<span><span class="co">#&gt; 10%   </span></span>
<span><span class="co">#&gt; 11%   </span></span>
<span><span class="co">#&gt; 11%   </span></span>
<span><span class="co">#&gt; 11%   </span></span>
<span><span class="co">#&gt; 11%   </span></span>
<span><span class="co">#&gt; 11%   </span></span>
<span><span class="co">#&gt; 11%   </span></span>
<span><span class="co">#&gt; 11%   </span></span>
<span><span class="co">#&gt; 11%   </span></span>
<span><span class="co">#&gt; 11%   </span></span>
<span><span class="co">#&gt; 12%   </span></span>
<span><span class="co">#&gt; 12%   </span></span>
<span><span class="co">#&gt; 12%   </span></span>
<span><span class="co">#&gt; 12%   </span></span>
<span><span class="co">#&gt; 12%   </span></span>
<span><span class="co">#&gt; 12%   </span></span>
<span><span class="co">#&gt; 12%   </span></span>
<span><span class="co">#&gt; 12%   </span></span>
<span><span class="co">#&gt; 12%   </span></span>
<span><span class="co">#&gt; 12%   </span></span>
<span><span class="co">#&gt; 12%   </span></span>
<span><span class="co">#&gt; 13%   </span></span>
<span><span class="co">#&gt; 13%   </span></span>
<span><span class="co">#&gt; 13%   </span></span>
<span><span class="co">#&gt; 13%   </span></span>
<span><span class="co">#&gt; 13%   </span></span>
<span><span class="co">#&gt; 13%   </span></span>
<span><span class="co">#&gt; 13%   </span></span>
<span><span class="co">#&gt; 13%   </span></span>
<span><span class="co">#&gt; 13%   </span></span>
<span><span class="co">#&gt; 14%   </span></span>
<span><span class="co">#&gt; 14%   </span></span>
<span><span class="co">#&gt; 14%   </span></span>
<span><span class="co">#&gt; 14%   </span></span>
<span><span class="co">#&gt; 14%   </span></span>
<span><span class="co">#&gt; 14%   </span></span>
<span><span class="co">#&gt; 14%   </span></span>
<span><span class="co">#&gt; 14%   </span></span>
<span><span class="co">#&gt; 14%   </span></span>
<span><span class="co">#&gt; 14%   </span></span>
<span><span class="co">#&gt; 14%   </span></span>
<span><span class="co">#&gt; 15%   </span></span>
<span><span class="co">#&gt; 15%   </span></span>
<span><span class="co">#&gt; 15%   </span></span>
<span><span class="co">#&gt; 15%   </span></span>
<span><span class="co">#&gt; 15%   </span></span>
<span><span class="co">#&gt; 15%   </span></span>
<span><span class="co">#&gt; 15%   </span></span>
<span><span class="co">#&gt; 15%   </span></span>
<span><span class="co">#&gt; 15%   </span></span>
<span><span class="co">#&gt; 16%   </span></span>
<span><span class="co">#&gt; 16%   </span></span>
<span><span class="co">#&gt; 16%   </span></span>
<span><span class="co">#&gt; 16%   </span></span>
<span><span class="co">#&gt; 16%   </span></span>
<span><span class="co">#&gt; 16%   </span></span>
<span><span class="co">#&gt; 16%   </span></span>
<span><span class="co">#&gt; 16%   </span></span>
<span><span class="co">#&gt; 16%   </span></span>
<span><span class="co">#&gt; 16%   </span></span>
<span><span class="co">#&gt; 16%   </span></span>
<span><span class="co">#&gt; 17%   </span></span>
<span><span class="co">#&gt; 17%   </span></span>
<span><span class="co">#&gt; 17%   </span></span>
<span><span class="co">#&gt; 17%   </span></span>
<span><span class="co">#&gt; 17%   </span></span>
<span><span class="co">#&gt; 17%   </span></span>
<span><span class="co">#&gt; 17%   </span></span>
<span><span class="co">#&gt; 17%   </span></span>
<span><span class="co">#&gt; 17%   </span></span>
<span><span class="co">#&gt; 18%   </span></span>
<span><span class="co">#&gt; 18%   </span></span>
<span><span class="co">#&gt; 18%   </span></span>
<span><span class="co">#&gt; 18%   </span></span>
<span><span class="co">#&gt; 18%   </span></span>
<span><span class="co">#&gt; 18%   </span></span>
<span><span class="co">#&gt; 18%   </span></span>
<span><span class="co">#&gt; 18%   </span></span>
<span><span class="co">#&gt; 18%   </span></span>
<span><span class="co">#&gt; 18%   </span></span>
<span><span class="co">#&gt; 18%   </span></span>
<span><span class="co">#&gt; 19%   </span></span>
<span><span class="co">#&gt; 19%   </span></span>
<span><span class="co">#&gt; 19%   </span></span>
<span><span class="co">#&gt; 19%   </span></span>
<span><span class="co">#&gt; 19%   </span></span>
<span><span class="co">#&gt; 19%   </span></span>
<span><span class="co">#&gt; 19%   </span></span>
<span><span class="co">#&gt; 19%   </span></span>
<span><span class="co">#&gt; 19%   </span></span>
<span><span class="co">#&gt; 20%   </span></span>
<span><span class="co">#&gt; 20%   </span></span>
<span><span class="co">#&gt; 20%   </span></span>
<span><span class="co">#&gt; 20%   </span></span>
<span><span class="co">#&gt; 20%   </span></span>
<span><span class="co">#&gt; 20%   </span></span>
<span><span class="co">#&gt; 20%   </span></span>
<span><span class="co">#&gt; 20%   </span></span>
<span><span class="co">#&gt; 20%   </span></span>
<span><span class="co">#&gt; 20%   </span></span>
<span><span class="co">#&gt; 20%   </span></span>
<span><span class="co">#&gt; 21%   </span></span>
<span><span class="co">#&gt; 21%   </span></span>
<span><span class="co">#&gt; 21%   </span></span>
<span><span class="co">#&gt; 21%   </span></span>
<span><span class="co">#&gt; 21%   </span></span>
<span><span class="co">#&gt; 21%   </span></span>
<span><span class="co">#&gt; 21%   </span></span>
<span><span class="co">#&gt; 21%   </span></span>
<span><span class="co">#&gt; 21%   </span></span>
<span><span class="co">#&gt; 22%   </span></span>
<span><span class="co">#&gt; 22%   </span></span>
<span><span class="co">#&gt; 22%   </span></span>
<span><span class="co">#&gt; 22%   </span></span>
<span><span class="co">#&gt; 22%   </span></span>
<span><span class="co">#&gt; 22%   </span></span>
<span><span class="co">#&gt; 22%   </span></span>
<span><span class="co">#&gt; 22%   </span></span>
<span><span class="co">#&gt; 22%   </span></span>
<span><span class="co">#&gt; 22%   </span></span>
<span><span class="co">#&gt; 22%   </span></span>
<span><span class="co">#&gt; 23%   </span></span>
<span><span class="co">#&gt; 23%   </span></span>
<span><span class="co">#&gt; 23%   </span></span>
<span><span class="co">#&gt; 23%   </span></span>
<span><span class="co">#&gt; 23%   </span></span>
<span><span class="co">#&gt; 23%   </span></span>
<span><span class="co">#&gt; 23%   </span></span>
<span><span class="co">#&gt; 23%   </span></span>
<span><span class="co">#&gt; 23%   </span></span>
<span><span class="co">#&gt; 24%   </span></span>
<span><span class="co">#&gt; 24%   </span></span>
<span><span class="co">#&gt; 24%   </span></span>
<span><span class="co">#&gt; 24%   </span></span>
<span><span class="co">#&gt; 24%   </span></span>
<span><span class="co">#&gt; 24%   </span></span>
<span><span class="co">#&gt; 24%   </span></span>
<span><span class="co">#&gt; 24%   </span></span>
<span><span class="co">#&gt; 24%   </span></span>
<span><span class="co">#&gt; 24%   </span></span>
<span><span class="co">#&gt; 24%   </span></span>
<span><span class="co">#&gt; 25%   </span></span>
<span><span class="co">#&gt; 25%   </span></span>
<span><span class="co">#&gt; 25%   </span></span>
<span><span class="co">#&gt; 25%   </span></span>
<span><span class="co">#&gt; 25%   </span></span>
<span><span class="co">#&gt; 25%   </span></span>
<span><span class="co">#&gt; 25%   </span></span>
<span><span class="co">#&gt; 25%   </span></span>
<span><span class="co">#&gt; 25%   </span></span>
<span><span class="co">#&gt; 26%   </span></span>
<span><span class="co">#&gt; 26%   </span></span>
<span><span class="co">#&gt; 26%   </span></span>
<span><span class="co">#&gt; 26%   </span></span>
<span><span class="co">#&gt; 26%   </span></span>
<span><span class="co">#&gt; 26%   </span></span>
<span><span class="co">#&gt; 26%   </span></span>
<span><span class="co">#&gt; 26%   </span></span>
<span><span class="co">#&gt; 26%   </span></span>
<span><span class="co">#&gt; 26%   </span></span>
<span><span class="co">#&gt; 26%   </span></span>
<span><span class="co">#&gt; 27%   </span></span>
<span><span class="co">#&gt; 27%   </span></span>
<span><span class="co">#&gt; 27%   </span></span>
<span><span class="co">#&gt; 27%   </span></span>
<span><span class="co">#&gt; 27%   </span></span>
<span><span class="co">#&gt; 27%   </span></span>
<span><span class="co">#&gt; 27%   </span></span>
<span><span class="co">#&gt; 27%   </span></span>
<span><span class="co">#&gt; 27%   </span></span>
<span><span class="co">#&gt; 28%   </span></span>
<span><span class="co">#&gt; 28%   </span></span>
<span><span class="co">#&gt; 28%   </span></span>
<span><span class="co">#&gt; 28%   </span></span>
<span><span class="co">#&gt; 28%   </span></span>
<span><span class="co">#&gt; 28%   </span></span>
<span><span class="co">#&gt; 28%   </span></span>
<span><span class="co">#&gt; 28%   </span></span>
<span><span class="co">#&gt; 28%   </span></span>
<span><span class="co">#&gt; 28%   </span></span>
<span><span class="co">#&gt; 28%   </span></span>
<span><span class="co">#&gt; 29%   </span></span>
<span><span class="co">#&gt; 29%   </span></span>
<span><span class="co">#&gt; 29%   </span></span>
<span><span class="co">#&gt; 29%   </span></span>
<span><span class="co">#&gt; 29%   </span></span>
<span><span class="co">#&gt; 29%   </span></span>
<span><span class="co">#&gt; 29%   </span></span>
<span><span class="co">#&gt; 29%   </span></span>
<span><span class="co">#&gt; 29%   </span></span>
<span><span class="co">#&gt; 30%   </span></span>
<span><span class="co">#&gt; 30%   </span></span>
<span><span class="co">#&gt; 30%   </span></span>
<span><span class="co">#&gt; 30%   </span></span>
<span><span class="co">#&gt; 30%   </span></span>
<span><span class="co">#&gt; 30%   </span></span>
<span><span class="co">#&gt; 30%   </span></span>
<span><span class="co">#&gt; 30%   </span></span>
<span><span class="co">#&gt; 30%   </span></span>
<span><span class="co">#&gt; 30%   </span></span>
<span><span class="co">#&gt; 30%   </span></span>
<span><span class="co">#&gt; 31%   </span></span>
<span><span class="co">#&gt; 31%   </span></span>
<span><span class="co">#&gt; 31%   </span></span>
<span><span class="co">#&gt; 31%   </span></span>
<span><span class="co">#&gt; 31%   </span></span>
<span><span class="co">#&gt; 31%   </span></span>
<span><span class="co">#&gt; 31%   </span></span>
<span><span class="co">#&gt; 31%   </span></span>
<span><span class="co">#&gt; 31%   </span></span>
<span><span class="co">#&gt; 32%   </span></span>
<span><span class="co">#&gt; 32%   </span></span>
<span><span class="co">#&gt; 32%   </span></span>
<span><span class="co">#&gt; 32%   </span></span>
<span><span class="co">#&gt; 32%   </span></span>
<span><span class="co">#&gt; 32%   </span></span>
<span><span class="co">#&gt; 32%   </span></span>
<span><span class="co">#&gt; 32%   </span></span>
<span><span class="co">#&gt; 32%   </span></span>
<span><span class="co">#&gt; 32%   </span></span>
<span><span class="co">#&gt; 32%   </span></span>
<span><span class="co">#&gt; 33%   </span></span>
<span><span class="co">#&gt; 33%   </span></span>
<span><span class="co">#&gt; 33%   </span></span>
<span><span class="co">#&gt; 33%   </span></span>
<span><span class="co">#&gt; 33%   </span></span>
<span><span class="co">#&gt; 33%   </span></span>
<span><span class="co">#&gt; 33%   </span></span>
<span><span class="co">#&gt; 33%   </span></span>
<span><span class="co">#&gt; 33%   </span></span>
<span><span class="co">#&gt; 34%   </span></span>
<span><span class="co">#&gt; 34%   </span></span>
<span><span class="co">#&gt; 34%   </span></span>
<span><span class="co">#&gt; 34%   </span></span>
<span><span class="co">#&gt; 34%   </span></span>
<span><span class="co">#&gt; 34%   </span></span>
<span><span class="co">#&gt; 34%   </span></span>
<span><span class="co">#&gt; 34%   </span></span>
<span><span class="co">#&gt; 34%   </span></span>
<span><span class="co">#&gt; 34%   </span></span>
<span><span class="co">#&gt; 34%   </span></span>
<span><span class="co">#&gt; 35%   </span></span>
<span><span class="co">#&gt; 35%   </span></span>
<span><span class="co">#&gt; 35%   </span></span>
<span><span class="co">#&gt; 35%   </span></span>
<span><span class="co">#&gt; 35%   </span></span>
<span><span class="co">#&gt; 35%   </span></span>
<span><span class="co">#&gt; 35%   </span></span>
<span><span class="co">#&gt; 35%   </span></span>
<span><span class="co">#&gt; 35%   </span></span>
<span><span class="co">#&gt; 36%   </span></span>
<span><span class="co">#&gt; 36%   </span></span>
<span><span class="co">#&gt; 36%   </span></span>
<span><span class="co">#&gt; 36%   </span></span>
<span><span class="co">#&gt; 36%   </span></span>
<span><span class="co">#&gt; 36%   </span></span>
<span><span class="co">#&gt; 36%   </span></span>
<span><span class="co">#&gt; 36%   </span></span>
<span><span class="co">#&gt; 36%   </span></span>
<span><span class="co">#&gt; 36%   </span></span>
<span><span class="co">#&gt; 36%   </span></span>
<span><span class="co">#&gt; 37%   </span></span>
<span><span class="co">#&gt; 37%   </span></span>
<span><span class="co">#&gt; 37%   </span></span>
<span><span class="co">#&gt; 37%   </span></span>
<span><span class="co">#&gt; 37%   </span></span>
<span><span class="co">#&gt; 37%   </span></span>
<span><span class="co">#&gt; 37%   </span></span>
<span><span class="co">#&gt; 37%   </span></span>
<span><span class="co">#&gt; 37%   </span></span>
<span><span class="co">#&gt; 38%   </span></span>
<span><span class="co">#&gt; 38%   </span></span>
<span><span class="co">#&gt; 38%   </span></span>
<span><span class="co">#&gt; 38%   </span></span>
<span><span class="co">#&gt; 38%   </span></span>
<span><span class="co">#&gt; 38%   </span></span>
<span><span class="co">#&gt; 38%   </span></span>
<span><span class="co">#&gt; 38%   </span></span>
<span><span class="co">#&gt; 38%   </span></span>
<span><span class="co">#&gt; 38%   </span></span>
<span><span class="co">#&gt; 38%   </span></span>
<span><span class="co">#&gt; 39%   </span></span>
<span><span class="co">#&gt; 39%   </span></span>
<span><span class="co">#&gt; 39%   </span></span>
<span><span class="co">#&gt; 39%   </span></span>
<span><span class="co">#&gt; 39%   </span></span>
<span><span class="co">#&gt; 39%   </span></span>
<span><span class="co">#&gt; 39%   </span></span>
<span><span class="co">#&gt; 39%   </span></span>
<span><span class="co">#&gt; 39%   </span></span>
<span><span class="co">#&gt; 40%   </span></span>
<span><span class="co">#&gt; 40%   </span></span>
<span><span class="co">#&gt; 40%   </span></span>
<span><span class="co">#&gt; 40%   </span></span>
<span><span class="co">#&gt; 40%   </span></span>
<span><span class="co">#&gt; 40%   </span></span>
<span><span class="co">#&gt; 40%   </span></span>
<span><span class="co">#&gt; 40%   </span></span>
<span><span class="co">#&gt; 40%   </span></span>
<span><span class="co">#&gt; 40%   </span></span>
<span><span class="co">#&gt; 40%   </span></span>
<span><span class="co">#&gt; 41%   </span></span>
<span><span class="co">#&gt; 41%   </span></span>
<span><span class="co">#&gt; 41%   </span></span>
<span><span class="co">#&gt; 41%   </span></span>
<span><span class="co">#&gt; 41%   </span></span>
<span><span class="co">#&gt; 41%   </span></span>
<span><span class="co">#&gt; 41%   </span></span>
<span><span class="co">#&gt; 41%   </span></span>
<span><span class="co">#&gt; 41%   </span></span>
<span><span class="co">#&gt; 42%   </span></span>
<span><span class="co">#&gt; 42%   </span></span>
<span><span class="co">#&gt; 42%   </span></span>
<span><span class="co">#&gt; 42%   </span></span>
<span><span class="co">#&gt; 42%   </span></span>
<span><span class="co">#&gt; 42%   </span></span>
<span><span class="co">#&gt; 42%   </span></span>
<span><span class="co">#&gt; 42%   </span></span>
<span><span class="co">#&gt; 42%   </span></span>
<span><span class="co">#&gt; 42%   </span></span>
<span><span class="co">#&gt; 42%   </span></span>
<span><span class="co">#&gt; 43%   </span></span>
<span><span class="co">#&gt; 43%   </span></span>
<span><span class="co">#&gt; 43%   </span></span>
<span><span class="co">#&gt; 43%   </span></span>
<span><span class="co">#&gt; 43%   </span></span>
<span><span class="co">#&gt; 43%   </span></span>
<span><span class="co">#&gt; 43%   </span></span>
<span><span class="co">#&gt; 43%   </span></span>
<span><span class="co">#&gt; 43%   </span></span>
<span><span class="co">#&gt; 44%   </span></span>
<span><span class="co">#&gt; 44%   </span></span>
<span><span class="co">#&gt; 44%   </span></span>
<span><span class="co">#&gt; 44%   </span></span>
<span><span class="co">#&gt; 44%   </span></span>
<span><span class="co">#&gt; 44%   </span></span>
<span><span class="co">#&gt; 44%   </span></span>
<span><span class="co">#&gt; 44%   </span></span>
<span><span class="co">#&gt; 44%   </span></span>
<span><span class="co">#&gt; 44%   </span></span>
<span><span class="co">#&gt; 44%   </span></span>
<span><span class="co">#&gt; 45%   </span></span>
<span><span class="co">#&gt; 45%   </span></span>
<span><span class="co">#&gt; 45%   </span></span>
<span><span class="co">#&gt; 45%   </span></span>
<span><span class="co">#&gt; 45%   </span></span>
<span><span class="co">#&gt; 45%   </span></span>
<span><span class="co">#&gt; 45%   </span></span>
<span><span class="co">#&gt; 45%   </span></span>
<span><span class="co">#&gt; 45%   </span></span>
<span><span class="co">#&gt; 46%   </span></span>
<span><span class="co">#&gt; 46%   </span></span>
<span><span class="co">#&gt; 46%   </span></span>
<span><span class="co">#&gt; 46%   </span></span>
<span><span class="co">#&gt; 46%   </span></span>
<span><span class="co">#&gt; 46%   </span></span>
<span><span class="co">#&gt; 46%   </span></span>
<span><span class="co">#&gt; 46%   </span></span>
<span><span class="co">#&gt; 46%   </span></span>
<span><span class="co">#&gt; 46%   </span></span>
<span><span class="co">#&gt; 46%   </span></span>
<span><span class="co">#&gt; 47%   </span></span>
<span><span class="co">#&gt; 47%   </span></span>
<span><span class="co">#&gt; 47%   </span></span>
<span><span class="co">#&gt; 47%   </span></span>
<span><span class="co">#&gt; 47%   </span></span>
<span><span class="co">#&gt; 47%   </span></span>
<span><span class="co">#&gt; 47%   </span></span>
<span><span class="co">#&gt; 47%   </span></span>
<span><span class="co">#&gt; 47%   </span></span>
<span><span class="co">#&gt; 48%   </span></span>
<span><span class="co">#&gt; 48%   </span></span>
<span><span class="co">#&gt; 48%   </span></span>
<span><span class="co">#&gt; 48%   </span></span>
<span><span class="co">#&gt; 48%   </span></span>
<span><span class="co">#&gt; 48%   </span></span>
<span><span class="co">#&gt; 48%   </span></span>
<span><span class="co">#&gt; 48%   </span></span>
<span><span class="co">#&gt; 48%   </span></span>
<span><span class="co">#&gt; 48%   </span></span>
<span><span class="co">#&gt; 48%   </span></span>
<span><span class="co">#&gt; 49%   </span></span>
<span><span class="co">#&gt; 49%   </span></span>
<span><span class="co">#&gt; 49%   </span></span>
<span><span class="co">#&gt; 49%   </span></span>
<span><span class="co">#&gt; 49%   </span></span>
<span><span class="co">#&gt; 49%   </span></span>
<span><span class="co">#&gt; 49%   </span></span>
<span><span class="co">#&gt; 49%   </span></span>
<span><span class="co">#&gt; 49%   </span></span>
<span><span class="co">#&gt; 50%   </span></span>
<span><span class="co">#&gt; 50%   </span></span>
<span><span class="co">#&gt; 50%   </span></span>
<span><span class="co">#&gt; 50%   </span></span>
<span><span class="co">#&gt; 50%   </span></span>
<span><span class="co">#&gt; 50%   </span></span>
<span><span class="co">#&gt; 50%   </span></span>
<span><span class="co">#&gt; 50%   </span></span>
<span><span class="co">#&gt; 50%   </span></span>
<span><span class="co">#&gt; 50%   </span></span>
<span><span class="co">#&gt; 50%   </span></span>
<span><span class="co">#&gt; 51%   </span></span>
<span><span class="co">#&gt; 51%   </span></span>
<span><span class="co">#&gt; 51%   </span></span>
<span><span class="co">#&gt; 51%   </span></span>
<span><span class="co">#&gt; 51%   </span></span>
<span><span class="co">#&gt; 51%   </span></span>
<span><span class="co">#&gt; 51%   </span></span>
<span><span class="co">#&gt; 51%   </span></span>
<span><span class="co">#&gt; 51%   </span></span>
<span><span class="co">#&gt; 52%   </span></span>
<span><span class="co">#&gt; 52%   </span></span>
<span><span class="co">#&gt; 52%   </span></span>
<span><span class="co">#&gt; 52%   </span></span>
<span><span class="co">#&gt; 52%   </span></span>
<span><span class="co">#&gt; 52%   </span></span>
<span><span class="co">#&gt; 52%   </span></span>
<span><span class="co">#&gt; 52%   </span></span>
<span><span class="co">#&gt; 52%   </span></span>
<span><span class="co">#&gt; 52%   </span></span>
<span><span class="co">#&gt; 52%   </span></span>
<span><span class="co">#&gt; 53%   </span></span>
<span><span class="co">#&gt; 53%   </span></span>
<span><span class="co">#&gt; 53%   </span></span>
<span><span class="co">#&gt; 53%   </span></span>
<span><span class="co">#&gt; 53%   </span></span>
<span><span class="co">#&gt; 53%   </span></span>
<span><span class="co">#&gt; 53%   </span></span>
<span><span class="co">#&gt; 53%   </span></span>
<span><span class="co">#&gt; 53%   </span></span>
<span><span class="co">#&gt; 54%   </span></span>
<span><span class="co">#&gt; 54%   </span></span>
<span><span class="co">#&gt; 54%   </span></span>
<span><span class="co">#&gt; 54%   </span></span>
<span><span class="co">#&gt; 54%   </span></span>
<span><span class="co">#&gt; 54%   </span></span>
<span><span class="co">#&gt; 54%   </span></span>
<span><span class="co">#&gt; 54%   </span></span>
<span><span class="co">#&gt; 54%   </span></span>
<span><span class="co">#&gt; 54%   </span></span>
<span><span class="co">#&gt; 55%   </span></span>
<span><span class="co">#&gt; 55%   </span></span>
<span><span class="co">#&gt; 55%   </span></span>
<span><span class="co">#&gt; 55%   </span></span>
<span><span class="co">#&gt; 55%   </span></span>
<span><span class="co">#&gt; 55%   </span></span>
<span><span class="co">#&gt; 55%   </span></span>
<span><span class="co">#&gt; 55%   </span></span>
<span><span class="co">#&gt; 55%   </span></span>
<span><span class="co">#&gt; 55%   </span></span>
<span><span class="co">#&gt; 56%   </span></span>
<span><span class="co">#&gt; 56%   </span></span>
<span><span class="co">#&gt; 56%   </span></span>
<span><span class="co">#&gt; 56%   </span></span>
<span><span class="co">#&gt; 56%   </span></span>
<span><span class="co">#&gt; 56%   </span></span>
<span><span class="co">#&gt; 56%   </span></span>
<span><span class="co">#&gt; 56%   </span></span>
<span><span class="co">#&gt; 56%   </span></span>
<span><span class="co">#&gt; 56%   </span></span>
<span><span class="co">#&gt; 56%   </span></span>
<span><span class="co">#&gt; 57%   </span></span>
<span><span class="co">#&gt; 57%   </span></span>
<span><span class="co">#&gt; 57%   </span></span>
<span><span class="co">#&gt; 57%   </span></span>
<span><span class="co">#&gt; 57%   </span></span>
<span><span class="co">#&gt; 57%   </span></span>
<span><span class="co">#&gt; 57%   </span></span>
<span><span class="co">#&gt; 57%   </span></span>
<span><span class="co">#&gt; 57%   </span></span>
<span><span class="co">#&gt; 57%   </span></span>
<span><span class="co">#&gt; 58%   </span></span>
<span><span class="co">#&gt; 58%   </span></span>
<span><span class="co">#&gt; 58%   </span></span>
<span><span class="co">#&gt; 58%   </span></span>
<span><span class="co">#&gt; 58%   </span></span>
<span><span class="co">#&gt; 58%   </span></span>
<span><span class="co">#&gt; 58%   </span></span>
<span><span class="co">#&gt; 58%   </span></span>
<span><span class="co">#&gt; 58%   </span></span>
<span><span class="co">#&gt; 58%   </span></span>
<span><span class="co">#&gt; 59%   </span></span>
<span><span class="co">#&gt; 59%   </span></span>
<span><span class="co">#&gt; 59%   </span></span>
<span><span class="co">#&gt; 59%   </span></span>
<span><span class="co">#&gt; 59%   </span></span>
<span><span class="co">#&gt; 59%   </span></span>
<span><span class="co">#&gt; 59%   </span></span>
<span><span class="co">#&gt; 59%   </span></span>
<span><span class="co">#&gt; 59%   </span></span>
<span><span class="co">#&gt; 60%   </span></span>
<span><span class="co">#&gt; 60%   </span></span>
<span><span class="co">#&gt; 60%   </span></span>
<span><span class="co">#&gt; 60%   </span></span>
<span><span class="co">#&gt; 60%   </span></span>
<span><span class="co">#&gt; 60%   </span></span>
<span><span class="co">#&gt; 60%   </span></span>
<span><span class="co">#&gt; 60%   </span></span>
<span><span class="co">#&gt; 60%   </span></span>
<span><span class="co">#&gt; 60%   </span></span>
<span><span class="co">#&gt; 60%   </span></span>
<span><span class="co">#&gt; 61%   </span></span>
<span><span class="co">#&gt; 61%   </span></span>
<span><span class="co">#&gt; 61%   </span></span>
<span><span class="co">#&gt; 61%   </span></span>
<span><span class="co">#&gt; 61%   </span></span>
<span><span class="co">#&gt; 61%   </span></span>
<span><span class="co">#&gt; 61%   </span></span>
<span><span class="co">#&gt; 61%   </span></span>
<span><span class="co">#&gt; 61%   </span></span>
<span><span class="co">#&gt; 62%   </span></span>
<span><span class="co">#&gt; 62%   </span></span>
<span><span class="co">#&gt; 62%   </span></span>
<span><span class="co">#&gt; 62%   </span></span>
<span><span class="co">#&gt; 62%   </span></span>
<span><span class="co">#&gt; 62%   </span></span>
<span><span class="co">#&gt; 62%   </span></span>
<span><span class="co">#&gt; 62%   </span></span>
<span><span class="co">#&gt; 62%   </span></span>
<span><span class="co">#&gt; 62%   </span></span>
<span><span class="co">#&gt; 62%   </span></span>
<span><span class="co">#&gt; 63%   </span></span>
<span><span class="co">#&gt; 63%   </span></span>
<span><span class="co">#&gt; 63%   </span></span>
<span><span class="co">#&gt; 63%   </span></span>
<span><span class="co">#&gt; 63%   </span></span>
<span><span class="co">#&gt; 63%   </span></span>
<span><span class="co">#&gt; 63%   </span></span>
<span><span class="co">#&gt; 63%   </span></span>
<span><span class="co">#&gt; 63%   </span></span>
<span><span class="co">#&gt; 64%   </span></span>
<span><span class="co">#&gt; 64%   </span></span>
<span><span class="co">#&gt; 64%   </span></span>
<span><span class="co">#&gt; 64%   </span></span>
<span><span class="co">#&gt; 64%   </span></span>
<span><span class="co">#&gt; 64%   </span></span>
<span><span class="co">#&gt; 64%   </span></span>
<span><span class="co">#&gt; 64%   </span></span>
<span><span class="co">#&gt; 64%   </span></span>
<span><span class="co">#&gt; 64%   </span></span>
<span><span class="co">#&gt; 64%   </span></span>
<span><span class="co">#&gt; 65%   </span></span>
<span><span class="co">#&gt; 65%   </span></span>
<span><span class="co">#&gt; 65%   </span></span>
<span><span class="co">#&gt; 65%   </span></span>
<span><span class="co">#&gt; 65%   </span></span>
<span><span class="co">#&gt; 65%   </span></span>
<span><span class="co">#&gt; 65%   </span></span>
<span><span class="co">#&gt; 65%   </span></span>
<span><span class="co">#&gt; 65%   </span></span>
<span><span class="co">#&gt; 66%   </span></span>
<span><span class="co">#&gt; 66%   </span></span>
<span><span class="co">#&gt; 66%   </span></span>
<span><span class="co">#&gt; 66%   </span></span>
<span><span class="co">#&gt; 66%   </span></span>
<span><span class="co">#&gt; 66%   </span></span>
<span><span class="co">#&gt; 66%   </span></span>
<span><span class="co">#&gt; 66%   </span></span>
<span><span class="co">#&gt; 66%   </span></span>
<span><span class="co">#&gt; 66%   </span></span>
<span><span class="co">#&gt; 66%   </span></span>
<span><span class="co">#&gt; 67%   </span></span>
<span><span class="co">#&gt; 67%   </span></span>
<span><span class="co">#&gt; 67%   </span></span>
<span><span class="co">#&gt; 67%   </span></span>
<span><span class="co">#&gt; 67%   </span></span>
<span><span class="co">#&gt; 67%   </span></span>
<span><span class="co">#&gt; 67%   </span></span>
<span><span class="co">#&gt; 67%   </span></span>
<span><span class="co">#&gt; 67%   </span></span>
<span><span class="co">#&gt; 68%   </span></span>
<span><span class="co">#&gt; 68%   </span></span>
<span><span class="co">#&gt; 68%   </span></span>
<span><span class="co">#&gt; 68%   </span></span>
<span><span class="co">#&gt; 68%   </span></span>
<span><span class="co">#&gt; 68%   </span></span>
<span><span class="co">#&gt; 68%   </span></span>
<span><span class="co">#&gt; 68%   </span></span>
<span><span class="co">#&gt; 68%   </span></span>
<span><span class="co">#&gt; 68%   </span></span>
<span><span class="co">#&gt; 68%   </span></span>
<span><span class="co">#&gt; 69%   </span></span>
<span><span class="co">#&gt; 69%   </span></span>
<span><span class="co">#&gt; 69%   </span></span>
<span><span class="co">#&gt; 69%   </span></span>
<span><span class="co">#&gt; 69%   </span></span>
<span><span class="co">#&gt; 69%   </span></span>
<span><span class="co">#&gt; 69%   </span></span>
<span><span class="co">#&gt; 69%   </span></span>
<span><span class="co">#&gt; 69%   </span></span>
<span><span class="co">#&gt; 70%   </span></span>
<span><span class="co">#&gt; 70%   </span></span>
<span><span class="co">#&gt; 70%   </span></span>
<span><span class="co">#&gt; 70%   </span></span>
<span><span class="co">#&gt; 70%   </span></span>
<span><span class="co">#&gt; 70%   </span></span>
<span><span class="co">#&gt; 70%   </span></span>
<span><span class="co">#&gt; 70%   </span></span>
<span><span class="co">#&gt; 70%   </span></span>
<span><span class="co">#&gt; 70%   </span></span>
<span><span class="co">#&gt; 70%   </span></span>
<span><span class="co">#&gt; 71%   </span></span>
<span><span class="co">#&gt; 71%   </span></span>
<span><span class="co">#&gt; 71%   </span></span>
<span><span class="co">#&gt; 71%   </span></span>
<span><span class="co">#&gt; 71%   </span></span>
<span><span class="co">#&gt; 71%   </span></span>
<span><span class="co">#&gt; 71%   </span></span>
<span><span class="co">#&gt; 71%   </span></span>
<span><span class="co">#&gt; 71%   </span></span>
<span><span class="co">#&gt; 72%   </span></span>
<span><span class="co">#&gt; 72%   </span></span>
<span><span class="co">#&gt; 72%   </span></span>
<span><span class="co">#&gt; 72%   </span></span>
<span><span class="co">#&gt; 72%   </span></span>
<span><span class="co">#&gt; 72%   </span></span>
<span><span class="co">#&gt; 72%   </span></span>
<span><span class="co">#&gt; 72%   </span></span>
<span><span class="co">#&gt; 72%   </span></span>
<span><span class="co">#&gt; 72%   </span></span>
<span><span class="co">#&gt; 72%   </span></span>
<span><span class="co">#&gt; 73%   </span></span>
<span><span class="co">#&gt; 73%   </span></span>
<span><span class="co">#&gt; 73%   </span></span>
<span><span class="co">#&gt; 73%   </span></span>
<span><span class="co">#&gt; 73%   </span></span>
<span><span class="co">#&gt; 73%   </span></span>
<span><span class="co">#&gt; 73%   </span></span>
<span><span class="co">#&gt; 73%   </span></span>
<span><span class="co">#&gt; 73%   </span></span>
<span><span class="co">#&gt; 74%   </span></span>
<span><span class="co">#&gt; 74%   </span></span>
<span><span class="co">#&gt; 74%   </span></span>
<span><span class="co">#&gt; 74%   </span></span>
<span><span class="co">#&gt; 74%   </span></span>
<span><span class="co">#&gt; 74%   </span></span>
<span><span class="co">#&gt; 74%   </span></span>
<span><span class="co">#&gt; 74%   </span></span>
<span><span class="co">#&gt; 74%   </span></span>
<span><span class="co">#&gt; 74%   </span></span>
<span><span class="co">#&gt; 74%   </span></span>
<span><span class="co">#&gt; 75%   </span></span>
<span><span class="co">#&gt; 75%   </span></span>
<span><span class="co">#&gt; 75%   </span></span>
<span><span class="co">#&gt; 75%   </span></span>
<span><span class="co">#&gt; 75%   </span></span>
<span><span class="co">#&gt; 75%   </span></span>
<span><span class="co">#&gt; 75%   </span></span>
<span><span class="co">#&gt; 75%   </span></span>
<span><span class="co">#&gt; 75%   </span></span>
<span><span class="co">#&gt; 76%   </span></span>
<span><span class="co">#&gt; 76%   </span></span>
<span><span class="co">#&gt; 76%   </span></span>
<span><span class="co">#&gt; 76%   </span></span>
<span><span class="co">#&gt; 76%   </span></span>
<span><span class="co">#&gt; 76%   </span></span>
<span><span class="co">#&gt; 76%   </span></span>
<span><span class="co">#&gt; 76%   </span></span>
<span><span class="co">#&gt; 76%   </span></span>
<span><span class="co">#&gt; 76%   </span></span>
<span><span class="co">#&gt; 76%   </span></span>
<span><span class="co">#&gt; 77%   </span></span>
<span><span class="co">#&gt; 77%   </span></span>
<span><span class="co">#&gt; 77%   </span></span>
<span><span class="co">#&gt; 77%   </span></span>
<span><span class="co">#&gt; 77%   </span></span>
<span><span class="co">#&gt; 77%   </span></span>
<span><span class="co">#&gt; 77%   </span></span>
<span><span class="co">#&gt; 77%   </span></span>
<span><span class="co">#&gt; 77%   </span></span>
<span><span class="co">#&gt; 78%   </span></span>
<span><span class="co">#&gt; 78%   </span></span>
<span><span class="co">#&gt; 78%   </span></span>
<span><span class="co">#&gt; 78%   </span></span>
<span><span class="co">#&gt; 78%   </span></span>
<span><span class="co">#&gt; 78%   </span></span>
<span><span class="co">#&gt; 78%   </span></span>
<span><span class="co">#&gt; 78%   </span></span>
<span><span class="co">#&gt; 78%   </span></span>
<span><span class="co">#&gt; 78%   </span></span>
<span><span class="co">#&gt; 78%   </span></span>
<span><span class="co">#&gt; 79%   </span></span>
<span><span class="co">#&gt; 79%   </span></span>
<span><span class="co">#&gt; 79%   </span></span>
<span><span class="co">#&gt; 79%   </span></span>
<span><span class="co">#&gt; 79%   </span></span>
<span><span class="co">#&gt; 79%   </span></span>
<span><span class="co">#&gt; 79%   </span></span>
<span><span class="co">#&gt; 79%   </span></span>
<span><span class="co">#&gt; 79%   </span></span>
<span><span class="co">#&gt; 80%   </span></span>
<span><span class="co">#&gt; 80%   </span></span>
<span><span class="co">#&gt; 80%   </span></span>
<span><span class="co">#&gt; 80%   </span></span>
<span><span class="co">#&gt; 80%   </span></span>
<span><span class="co">#&gt; 80%   </span></span>
<span><span class="co">#&gt; 80%   </span></span>
<span><span class="co">#&gt; 80%   </span></span>
<span><span class="co">#&gt; 80%   </span></span>
<span><span class="co">#&gt; 80%   </span></span>
<span><span class="co">#&gt; 80%   </span></span>
<span><span class="co">#&gt; 81%   </span></span>
<span><span class="co">#&gt; 81%   </span></span>
<span><span class="co">#&gt; 81%   </span></span>
<span><span class="co">#&gt; 81%   </span></span>
<span><span class="co">#&gt; 81%   </span></span>
<span><span class="co">#&gt; 81%   </span></span>
<span><span class="co">#&gt; 81%   </span></span>
<span><span class="co">#&gt; 81%   </span></span>
<span><span class="co">#&gt; 81%   </span></span>
<span><span class="co">#&gt; 82%   </span></span>
<span><span class="co">#&gt; 82%   </span></span>
<span><span class="co">#&gt; 82%   </span></span>
<span><span class="co">#&gt; 82%   </span></span>
<span><span class="co">#&gt; 82%   </span></span>
<span><span class="co">#&gt; 82%   </span></span>
<span><span class="co">#&gt; 82%   </span></span>
<span><span class="co">#&gt; 82%   </span></span>
<span><span class="co">#&gt; 82%   </span></span>
<span><span class="co">#&gt; 82%   </span></span>
<span><span class="co">#&gt; 82%   </span></span>
<span><span class="co">#&gt; 83%   </span></span>
<span><span class="co">#&gt; 83%   </span></span>
<span><span class="co">#&gt; 83%   </span></span>
<span><span class="co">#&gt; 83%   </span></span>
<span><span class="co">#&gt; 83%   </span></span>
<span><span class="co">#&gt; 83%   </span></span>
<span><span class="co">#&gt; 83%   </span></span>
<span><span class="co">#&gt; 83%   </span></span>
<span><span class="co">#&gt; 83%   </span></span>
<span><span class="co">#&gt; 84%   </span></span>
<span><span class="co">#&gt; 84%   </span></span>
<span><span class="co">#&gt; 84%   </span></span>
<span><span class="co">#&gt; 84%   </span></span>
<span><span class="co">#&gt; 84%   </span></span>
<span><span class="co">#&gt; 84%   </span></span>
<span><span class="co">#&gt; 84%   </span></span>
<span><span class="co">#&gt; 84%   </span></span>
<span><span class="co">#&gt; 84%   </span></span>
<span><span class="co">#&gt; 84%   </span></span>
<span><span class="co">#&gt; 84%   </span></span>
<span><span class="co">#&gt; 85%   </span></span>
<span><span class="co">#&gt; 85%   </span></span>
<span><span class="co">#&gt; 85%   </span></span>
<span><span class="co">#&gt; 85%   </span></span>
<span><span class="co">#&gt; 85%   </span></span>
<span><span class="co">#&gt; 85%   </span></span>
<span><span class="co">#&gt; 85%   </span></span>
<span><span class="co">#&gt; 85%   </span></span>
<span><span class="co">#&gt; 85%   </span></span>
<span><span class="co">#&gt; 86%   </span></span>
<span><span class="co">#&gt; 86%   </span></span>
<span><span class="co">#&gt; 86%   </span></span>
<span><span class="co">#&gt; 86%   </span></span>
<span><span class="co">#&gt; 86%   </span></span>
<span><span class="co">#&gt; 86%   </span></span>
<span><span class="co">#&gt; 86%   </span></span>
<span><span class="co">#&gt; 86%   </span></span>
<span><span class="co">#&gt; 86%   </span></span>
<span><span class="co">#&gt; 86%   </span></span>
<span><span class="co">#&gt; 86%   </span></span>
<span><span class="co">#&gt; 87%   </span></span>
<span><span class="co">#&gt; 87%   </span></span>
<span><span class="co">#&gt; 87%   </span></span>
<span><span class="co">#&gt; 87%   </span></span>
<span><span class="co">#&gt; 87%   </span></span>
<span><span class="co">#&gt; 87%   </span></span>
<span><span class="co">#&gt; 87%   </span></span>
<span><span class="co">#&gt; 87%   </span></span>
<span><span class="co">#&gt; 87%   </span></span>
<span><span class="co">#&gt; 88%   </span></span>
<span><span class="co">#&gt; 88%   </span></span>
<span><span class="co">#&gt; 88%   </span></span>
<span><span class="co">#&gt; 88%   </span></span>
<span><span class="co">#&gt; 88%   </span></span>
<span><span class="co">#&gt; 88%   </span></span>
<span><span class="co">#&gt; 88%   </span></span>
<span><span class="co">#&gt; 88%   </span></span>
<span><span class="co">#&gt; 88%   </span></span>
<span><span class="co">#&gt; 88%   </span></span>
<span><span class="co">#&gt; 88%   </span></span>
<span><span class="co">#&gt; 89%   </span></span>
<span><span class="co">#&gt; 89%   </span></span>
<span><span class="co">#&gt; 89%   </span></span>
<span><span class="co">#&gt; 89%   </span></span>
<span><span class="co">#&gt; 89%   </span></span>
<span><span class="co">#&gt; 89%   </span></span>
<span><span class="co">#&gt; 89%   </span></span>
<span><span class="co">#&gt; 89%   </span></span>
<span><span class="co">#&gt; 89%   </span></span>
<span><span class="co">#&gt; 90%   </span></span>
<span><span class="co">#&gt; 90%   </span></span>
<span><span class="co">#&gt; 90%   </span></span>
<span><span class="co">#&gt; 90%   </span></span>
<span><span class="co">#&gt; 90%   </span></span>
<span><span class="co">#&gt; 90%   </span></span>
<span><span class="co">#&gt; 90%   </span></span>
<span><span class="co">#&gt; 90%   </span></span>
<span><span class="co">#&gt; 90%   </span></span>
<span><span class="co">#&gt; 90%   </span></span>
<span><span class="co">#&gt; 90%   </span></span>
<span><span class="co">#&gt; 91%   </span></span>
<span><span class="co">#&gt; 91%   </span></span>
<span><span class="co">#&gt; 91%   </span></span>
<span><span class="co">#&gt; 91%   </span></span>
<span><span class="co">#&gt; 91%   </span></span>
<span><span class="co">#&gt; 91%   </span></span>
<span><span class="co">#&gt; 91%   </span></span>
<span><span class="co">#&gt; 91%   </span></span>
<span><span class="co">#&gt; 91%   </span></span>
<span><span class="co">#&gt; 92%   </span></span>
<span><span class="co">#&gt; 92%   </span></span>
<span><span class="co">#&gt; 92%   </span></span>
<span><span class="co">#&gt; 92%   </span></span>
<span><span class="co">#&gt; 92%   </span></span>
<span><span class="co">#&gt; 92%   </span></span>
<span><span class="co">#&gt; 92%   </span></span>
<span><span class="co">#&gt; 92%   </span></span>
<span><span class="co">#&gt; 92%   </span></span>
<span><span class="co">#&gt; 92%   </span></span>
<span><span class="co">#&gt; 92%   </span></span>
<span><span class="co">#&gt; 93%   </span></span>
<span><span class="co">#&gt; 93%   </span></span>
<span><span class="co">#&gt; 93%   </span></span>
<span><span class="co">#&gt; 93%   </span></span>
<span><span class="co">#&gt; 93%   </span></span>
<span><span class="co">#&gt; 93%   </span></span>
<span><span class="co">#&gt; 93%   </span></span>
<span><span class="co">#&gt; 93%   </span></span>
<span><span class="co">#&gt; 93%   </span></span>
<span><span class="co">#&gt; 94%   </span></span>
<span><span class="co">#&gt; 94%   </span></span>
<span><span class="co">#&gt; 94%   </span></span>
<span><span class="co">#&gt; 94%   </span></span>
<span><span class="co">#&gt; 94%   </span></span>
<span><span class="co">#&gt; 94%   </span></span>
<span><span class="co">#&gt; 94%   </span></span>
<span><span class="co">#&gt; 94%   </span></span>
<span><span class="co">#&gt; 94%   </span></span>
<span><span class="co">#&gt; 94%   </span></span>
<span><span class="co">#&gt; 94%   </span></span>
<span><span class="co">#&gt; 95%   </span></span>
<span><span class="co">#&gt; 95%   </span></span>
<span><span class="co">#&gt; 95%   </span></span>
<span><span class="co">#&gt; 95%   </span></span>
<span><span class="co">#&gt; 95%   </span></span>
<span><span class="co">#&gt; 95%   </span></span>
<span><span class="co">#&gt; 95%   </span></span>
<span><span class="co">#&gt; 95%   </span></span>
<span><span class="co">#&gt; 95%   </span></span>
<span><span class="co">#&gt; 96%   </span></span>
<span><span class="co">#&gt; 96%   </span></span>
<span><span class="co">#&gt; 96%   </span></span>
<span><span class="co">#&gt; 96%   </span></span>
<span><span class="co">#&gt; 96%   </span></span>
<span><span class="co">#&gt; 96%   </span></span>
<span><span class="co">#&gt; 96%   </span></span>
<span><span class="co">#&gt; 96%   </span></span>
<span><span class="co">#&gt; 96%   </span></span>
<span><span class="co">#&gt; 96%   </span></span>
<span><span class="co">#&gt; 96%   </span></span>
<span><span class="co">#&gt; 97%   </span></span>
<span><span class="co">#&gt; 97%   </span></span>
<span><span class="co">#&gt; 97%   </span></span>
<span><span class="co">#&gt; 97%   </span></span>
<span><span class="co">#&gt; 97%   </span></span>
<span><span class="co">#&gt; 97%   </span></span>
<span><span class="co">#&gt; 97%   </span></span>
<span><span class="co">#&gt; 97%   </span></span>
<span><span class="co">#&gt; 97%   </span></span>
<span><span class="co">#&gt; 98%   </span></span>
<span><span class="co">#&gt; 98%   </span></span>
<span><span class="co">#&gt; 98%   </span></span>
<span><span class="co">#&gt; 98%   </span></span>
<span><span class="co">#&gt; 98%   </span></span>
<span><span class="co">#&gt; 98%   </span></span>
<span><span class="co">#&gt; 98%   </span></span>
<span><span class="co">#&gt; 98%   </span></span>
<span><span class="co">#&gt; 98%   </span></span>
<span><span class="co">#&gt; 98%   </span></span>
<span><span class="co">#&gt; 98%   </span></span>
<span><span class="co">#&gt; 99%   </span></span>
<span><span class="co">#&gt; 99%   </span></span>
<span><span class="co">#&gt; 99%   </span></span>
<span><span class="co">#&gt; 99%   </span></span>
<span><span class="co">#&gt; 99%   </span></span>
<span><span class="co">#&gt; 99%   </span></span>
<span><span class="co">#&gt; 99%   </span></span>
<span><span class="co">#&gt; 99%   </span></span>
<span><span class="co">#&gt; 99%   </span></span>
<span><span class="co">#&gt; 100%   </span></span>
<span><span class="co">#&gt; 100%   </span></span>
<span><span class="co">#&gt; 100%   </span></span>
<span><span class="co">#&gt; 100%   </span></span>
<span><span class="co">#&gt; 100%   </span></span>
<span><span class="co">#&gt; 100%   </span></span>
<span><span class="co">#&gt; Calculating the p-values...</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">GO.samp</span><span class="op">)</span></span>
<span><span class="co">#&gt;       category over_represented_pvalue under_represented_pvalue numDEInCat numInCat</span></span>
<span><span class="co">#&gt; 33  GO:0000070             0.000999001                        1         58      176</span></span>
<span><span class="co">#&gt; 40  GO:0000086             0.000999001                        1         42      125</span></span>
<span><span class="co">#&gt; 100 GO:0000226             0.000999001                        1        150      560</span></span>
<span><span class="co">#&gt; 123 GO:0000278             0.000999001                        1        227      787</span></span>
<span><span class="co">#&gt; 124 GO:0000280             0.000999001                        1        109      374</span></span>
<span><span class="co">#&gt; 244 GO:0000727             0.000999001                        1          8       12</span></span>
<span><span class="co">#&gt;                                                         term ontology</span></span>
<span><span class="co">#&gt; 33                      mitotic sister chromatid segregation       BP</span></span>
<span><span class="co">#&gt; 40                     G2/M transition of mitotic cell cycle       BP</span></span>
<span><span class="co">#&gt; 100                    microtubule cytoskeleton organization       BP</span></span>
<span><span class="co">#&gt; 123                                       mitotic cell cycle       BP</span></span>
<span><span class="co">#&gt; 124                                         nuclear division       BP</span></span>
<span><span class="co">#&gt; 244 double-strand break repair via break-induced replication       BP</span></span></code></pre></div>
<p>You will notice that this takes far longer than the Wallenius
approximation. Plotting the p-values against one another, we see that
there is little difference between the two methods. However, the
accuracy of the sampling method is limited by the number of samples
generated, <code>repcnt</code>, such that very low p-values will not be
correctly calculated. Significantly enriched GO terms may then be missed
after correcting for multiple testing.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">GO.wall</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">GO.samp</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">GO.wall</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>, <span class="va">GO.samp</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span><span class="op">)</span>, <span class="fl">2</span><span class="op">]</span><span class="op">)</span>,</span>
<span>  xlab <span class="op">=</span> <span class="st">"log10(Wallenius p-values)"</span>, ylab <span class="op">=</span> <span class="st">"log10(Sampling p-values)"</span>,</span>
<span>  xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">3</span>, <span class="fl">0</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span>, col <span class="op">=</span> <span class="fl">3</span>, lty <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="goseq_manual_files/figure-html/plot_wal_v_samp-1.png" width="576"></p>
</div>
<div class="section level4">
<h4 id="ignoring-length-bias">Ignoring length bias<a class="anchor" aria-label="anchor" href="#ignoring-length-bias"></a>
</h4>
<p><em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em> also
allows for one to perform a GO analysis without correcting for RNA-seq
length bias. In practice, this is only useful for assessing the effect
of length bias on your results. You should NEVER use this option as your
final analysis. If length bias is truly not present in your data, <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em> will
produce a nearly flat PWF and no length bias correction will be applied
to your data and all methods will produce the same results.</p>
<p>However, if you still wish to ignore length bias in calculating GO
category enrichment, this is again accomplished using the
<code>method</code> option.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">GO.nobias</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/goseq.html">goseq</a></span><span class="op">(</span><span class="va">pwf</span>, <span class="st">"hg19"</span>, <span class="st">"ensGene"</span>, method <span class="op">=</span> <span class="st">"Hypergeometric"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Fetching GO annotations...</span></span>
<span><span class="co">#&gt; For 9343 genes, we could not find any categories. These genes will be excluded.</span></span>
<span><span class="co">#&gt; To force their use, please run with use_genes_without_cat=TRUE (see documentation).</span></span>
<span><span class="co">#&gt; This was the default behavior for version 1.15.1 and earlier.</span></span>
<span><span class="co">#&gt; Calculating the p-values...</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">GO.nobias</span><span class="op">)</span></span>
<span><span class="co">#&gt;         category over_represented_pvalue under_represented_pvalue numDEInCat</span></span>
<span><span class="co">#&gt; 2490  GO:0005737            3.450552e-11                        1       2025</span></span>
<span><span class="co">#&gt; 9061  GO:0035556            1.023208e-09                        1        512</span></span>
<span><span class="co">#&gt; 123   GO:0000278            1.357715e-09                        1        227</span></span>
<span><span class="co">#&gt; 17061 GO:0110165            7.815802e-09                        1       2646</span></span>
<span><span class="co">#&gt; 15209 GO:0071944            9.604453e-09                        1        852</span></span>
<span><span class="co">#&gt; 12092 GO:0048731            1.155265e-08                        1        666</span></span>
<span><span class="co">#&gt;       numInCat                              term ontology</span></span>
<span><span class="co">#&gt; 2490      9352                         cytoplasm       CC</span></span>
<span><span class="co">#&gt; 9061      2029 intracellular signal transduction       BP</span></span>
<span><span class="co">#&gt; 123        787                mitotic cell cycle       BP</span></span>
<span><span class="co">#&gt; 17061    12873        cellular anatomical entity       CC</span></span>
<span><span class="co">#&gt; 15209     3638                    cell periphery       CC</span></span>
<span><span class="co">#&gt; 12092     2769                system development       BP</span></span></code></pre></div>
<p>Ignoring length bias gives very different results from a length bias
corrected analysis.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">GO.wall</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">GO.nobias</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">GO.wall</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>, <span class="va">GO.nobias</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span><span class="op">)</span>, <span class="fl">2</span><span class="op">]</span><span class="op">)</span>,</span>
<span>  xlab <span class="op">=</span> <span class="st">"log10(Wallenius p-values)"</span>, ylab <span class="op">=</span> <span class="st">"log10(Hypergeometric p-values)"</span>,</span>
<span>  xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">3</span>, <span class="fl">0</span><span class="op">)</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">3</span>, <span class="fl">0</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span>, col <span class="op">=</span> <span class="fl">3</span>, lty <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="goseq_manual_files/figure-html/plot_wal_v_hyper-1.png" width="576"></p>
</div>
<div class="section level4">
<h4 id="limiting-go-categories-and-other-category-based-tests">Limiting GO categories and other category based tests<a class="anchor" aria-label="anchor" href="#limiting-go-categories-and-other-category-based-tests"></a>
</h4>
<p>By default, <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em> tests
all three major Gene Ontology branches: Cellular Components, Biological
Processes and Molecular Functions. However, it is possible to limit
testing to any combination of the major branches by using the
<code>test.cats</code> argument to the <code><a href="../reference/goseq.html">goseq()</a></code> function.
This is done by specifying a vector consisting of some combination of
the strings <code>GO:CC</code>, <code>GO:BP</code> and
<code>GO:MF</code>. For example, to test for only Molecular Function GO
categories:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">GO.MF</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/goseq.html">goseq</a></span><span class="op">(</span><span class="va">pwf</span>, <span class="st">"hg19"</span>, <span class="st">"ensGene"</span>, test.cats <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GO:MF"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; Fetching GO annotations...</span></span>
<span><span class="co">#&gt; For 9343 genes, we could not find any categories. These genes will be excluded.</span></span>
<span><span class="co">#&gt; To force their use, please run with use_genes_without_cat=TRUE (see documentation).</span></span>
<span><span class="co">#&gt; This was the default behavior for version 1.15.1 and earlier.</span></span>
<span><span class="co">#&gt; Calculating the p-values...</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span></code></pre></div>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">GO.MF</span><span class="op">)</span></span>
<span><span class="co">#&gt;        category over_represented_pvalue under_represented_pvalue numDEInCat</span></span>
<span><span class="co">#&gt; 1011 GO:0005198            1.213677e-05                0.9999927        135</span></span>
<span><span class="co">#&gt; 1174 GO:0008092            8.175368e-05                0.9999420        219</span></span>
<span><span class="co">#&gt; 228  GO:0003735            8.457175e-05                0.9999623         40</span></span>
<span><span class="co">#&gt; 1160 GO:0008017            1.229304e-04                0.9999309         78</span></span>
<span><span class="co">#&gt; 2267 GO:0030215            1.596166e-04                0.9999788         11</span></span>
<span><span class="co">#&gt; 746  GO:0004725            4.042999e-04                0.9998399         29</span></span>
<span><span class="co">#&gt;      numInCat                                  term ontology</span></span>
<span><span class="co">#&gt; 1011      519          structural molecule activity       MF</span></span>
<span><span class="co">#&gt; 1174      820          cytoskeletal protein binding       MF</span></span>
<span><span class="co">#&gt; 228       154    structural constituent of ribosome       MF</span></span>
<span><span class="co">#&gt; 1160      246                   microtubule binding       MF</span></span>
<span><span class="co">#&gt; 2267       17           semaphorin receptor binding       MF</span></span>
<span><span class="co">#&gt; 746        75 protein tyrosine phosphatase activity       MF</span></span></code></pre></div>
<p>Native support for other category tests, such as KEGG pathway
analysis are also made available via this argument. See the man
<code><a href="../reference/goseq.html">goseq()</a></code> function man page for up to date information on
what category tests are natively supported.</p>
</div>
<div class="section level4">
<h4 id="making-sense-of-the-results">Making sense of the results<a class="anchor" aria-label="anchor" href="#making-sense-of-the-results"></a>
</h4>
<p>Having performed the GO analysis, you may now wish to interpret the
results. If you wish to identify categories significantly
enriched/unenriched below some p-value cutoff, it is necessary to first
apply some kind of multiple hypothesis testing correction. For example,
GO categories over enriched using a .05 FDR cutoff <span class="citation">(Benjamini and Hochberg 1995)</span> are:</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">enriched.GO</span> <span class="op">&lt;-</span> </span>
<span>  <span class="va">GO.wall</span><span class="op">$</span><span class="va">category</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span><span class="va">GO.wall</span><span class="op">$</span><span class="va">over_represented_pvalue</span>, method <span class="op">=</span> <span class="st">"BH"</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">.05</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">enriched.GO</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "GO:0005737" "GO:0000278" "GO:0005615" "GO:0110165" "GO:0006793" "GO:0035556"</span></span></code></pre></div>
<p>Unless you are a machine, GO accession identifiers are probably not
very meaningful to you. Information about each term can be obtained from
the Gene Ontology website, <a href="http://www.geneontology.org/" class="external-link uri">http://www.geneontology.org/</a>, or using the R package
<em><a href="https://bioconductor.org/packages/3.19/GO.db" class="external-link">GO.db</a></em>.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"GO.db"</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">go</span> <span class="kw">in</span> <span class="va">enriched.GO</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">GOTERM</span><span class="op">[[</span><span class="va">go</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"--------------------------------------\n"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="co">#&gt; GOID: GO:0005737</span></span>
<span><span class="co">#&gt; Term: cytoplasm</span></span>
<span><span class="co">#&gt; Ontology: CC</span></span>
<span><span class="co">#&gt; Definition: The contents of a cell excluding the plasma membrane and</span></span>
<span><span class="co">#&gt;     nucleus, but including other subcellular structures.</span></span>
<span><span class="co">#&gt; --------------------------------------</span></span>
<span><span class="co">#&gt; GOID: GO:0000278</span></span>
<span><span class="co">#&gt; Term: mitotic cell cycle</span></span>
<span><span class="co">#&gt; Ontology: BP</span></span>
<span><span class="co">#&gt; Definition: Progression through the phases of the mitotic cell cycle, the</span></span>
<span><span class="co">#&gt;     most common eukaryotic cell cycle, which canonically comprises four</span></span>
<span><span class="co">#&gt;     successive phases called G1, S, G2, and M and includes replication of</span></span>
<span><span class="co">#&gt;     the genome and the subsequent segregation of chromosomes into daughter</span></span>
<span><span class="co">#&gt;     cells. In some variant cell cycles nuclear replication or nuclear</span></span>
<span><span class="co">#&gt;     division may not be followed by cell division, or G1 and G2 phases may</span></span>
<span><span class="co">#&gt;     be absent.</span></span>
<span><span class="co">#&gt; Synonym: GO:0007067</span></span>
<span><span class="co">#&gt; Synonym: mitosis</span></span>
<span><span class="co">#&gt; Secondary: GO:0007067</span></span>
<span><span class="co">#&gt; --------------------------------------</span></span>
<span><span class="co">#&gt; GOID: GO:0005615</span></span>
<span><span class="co">#&gt; Term: extracellular space</span></span>
<span><span class="co">#&gt; Ontology: CC</span></span>
<span><span class="co">#&gt; Definition: That part of a multicellular organism outside the cells</span></span>
<span><span class="co">#&gt;     proper, usually taken to be outside the plasma membranes, and occupied</span></span>
<span><span class="co">#&gt;     by fluid.</span></span>
<span><span class="co">#&gt; Synonym: intercellular space</span></span>
<span><span class="co">#&gt; --------------------------------------</span></span>
<span><span class="co">#&gt; GOID: GO:0110165</span></span>
<span><span class="co">#&gt; Term: cellular anatomical entity</span></span>
<span><span class="co">#&gt; Ontology: CC</span></span>
<span><span class="co">#&gt; Definition: A part of a cellular organism that is either an immaterial</span></span>
<span><span class="co">#&gt;     entity or a material entity with granularity above the level of a</span></span>
<span><span class="co">#&gt;     protein complex but below that of an anatomical system. Or, a</span></span>
<span><span class="co">#&gt;     substance produced by a cellular organism with granularity above the</span></span>
<span><span class="co">#&gt;     level of a protein complex.</span></span>
<span><span class="co">#&gt; --------------------------------------</span></span>
<span><span class="co">#&gt; GOID: GO:0006793</span></span>
<span><span class="co">#&gt; Term: phosphorus metabolic process</span></span>
<span><span class="co">#&gt; Ontology: BP</span></span>
<span><span class="co">#&gt; Definition: The chemical reactions and pathways involving the nonmetallic</span></span>
<span><span class="co">#&gt;     element phosphorus or compounds that contain phosphorus, usually in</span></span>
<span><span class="co">#&gt;     the form of a phosphate group (PO4).</span></span>
<span><span class="co">#&gt; Synonym: phosphorus metabolism</span></span>
<span><span class="co">#&gt; --------------------------------------</span></span>
<span><span class="co">#&gt; GOID: GO:0035556</span></span>
<span><span class="co">#&gt; Term: intracellular signal transduction</span></span>
<span><span class="co">#&gt; Ontology: BP</span></span>
<span><span class="co">#&gt; Definition: The process in which a signal is passed on to downstream</span></span>
<span><span class="co">#&gt;     components within the cell, which become activated themselves to</span></span>
<span><span class="co">#&gt;     further propagate the signal and finally trigger a change in the</span></span>
<span><span class="co">#&gt;     function or state of the cell.</span></span>
<span><span class="co">#&gt; Synonym: GO:0007242</span></span>
<span><span class="co">#&gt; Synonym: GO:0007243</span></span>
<span><span class="co">#&gt; Synonym: GO:0023013</span></span>
<span><span class="co">#&gt; Synonym: GO:0023034</span></span>
<span><span class="co">#&gt; Synonym: intracellular signaling cascade</span></span>
<span><span class="co">#&gt; Synonym: intracellular signaling pathway</span></span>
<span><span class="co">#&gt; Synonym: intracellular signal transduction pathway</span></span>
<span><span class="co">#&gt; Synonym: signal transmission via intracellular cascade</span></span>
<span><span class="co">#&gt; Secondary: GO:0007242</span></span>
<span><span class="co">#&gt; Secondary: GO:0007243</span></span>
<span><span class="co">#&gt; Secondary: GO:0023013</span></span>
<span><span class="co">#&gt; Secondary: GO:0023034</span></span>
<span><span class="co">#&gt; --------------------------------------</span></span>
<span><span class="co">#&gt; GOID: GO:0005575</span></span>
<span><span class="co">#&gt; Term: cellular_component</span></span>
<span><span class="co">#&gt; Ontology: CC</span></span>
<span><span class="co">#&gt; Definition: A location, relative to cellular compartments and structures,</span></span>
<span><span class="co">#&gt;     occupied by a macromolecular machine. There are three types of</span></span>
<span><span class="co">#&gt;     cellular components described in the gene ontology: (1) the cellular</span></span>
<span><span class="co">#&gt;     anatomical entity where a gene product carries out a molecular</span></span>
<span><span class="co">#&gt;     function (e.g., plasma membrane, cytoskeleton) or membrane-enclosed</span></span>
<span><span class="co">#&gt;     compartments (e.g., mitochondrion); (2) virion components, where viral</span></span>
<span><span class="co">#&gt;     proteins act, and (3) the stable macromolecular complexes of which</span></span>
<span><span class="co">#&gt;     gene product are parts (e.g., the clathrin complex).</span></span>
<span><span class="co">#&gt; Synonym: GO:0008372</span></span>
<span><span class="co">#&gt; Synonym: cell or subcellular entity</span></span>
<span><span class="co">#&gt; Synonym: cellular component</span></span>
<span><span class="co">#&gt; Synonym: subcellular entity</span></span>
<span><span class="co">#&gt; Secondary: GO:0008372</span></span>
<span><span class="co">#&gt; --------------------------------------</span></span>
<span><span class="co">#&gt; GOID: GO:0044281</span></span>
<span><span class="co">#&gt; Term: small molecule metabolic process</span></span>
<span><span class="co">#&gt; Ontology: BP</span></span>
<span><span class="co">#&gt; Definition: The chemical reactions and pathways involving small molecules,</span></span>
<span><span class="co">#&gt;     any low molecular weight, monomeric, non-encoded molecule.</span></span>
<span><span class="co">#&gt; Synonym: small molecule metabolism</span></span>
<span><span class="co">#&gt; --------------------------------------</span></span>
<span><span class="co">#&gt; GOID: GO:0051301</span></span>
<span><span class="co">#&gt; Term: cell division</span></span>
<span><span class="co">#&gt; Ontology: BP</span></span>
<span><span class="co">#&gt; Definition: The process resulting in division and partitioning of</span></span>
<span><span class="co">#&gt;     components of a cell to form more cells; may or may not be accompanied</span></span>
<span><span class="co">#&gt;     by the physical separation of a cell into distinct, individually</span></span>
<span><span class="co">#&gt;     membrane-bounded daughter cells.</span></span>
<span><span class="co">#&gt; --------------------------------------</span></span>
<span><span class="co">#&gt; GOID: GO:0070062</span></span>
<span><span class="co">#&gt; Term: extracellular exosome</span></span>
<span><span class="co">#&gt; Ontology: CC</span></span>
<span><span class="co">#&gt; Definition: A vesicle that is released into the extracellular region by</span></span>
<span><span class="co">#&gt;     fusion of the limiting endosomal membrane of a multivesicular body</span></span>
<span><span class="co">#&gt;     with the plasma membrane. Extracellular exosomes, also simply called</span></span>
<span><span class="co">#&gt;     exosomes, have a diameter of about 40-100 nm.</span></span>
<span><span class="co">#&gt; Synonym: exosome</span></span>
<span><span class="co">#&gt; Synonym: extracellular vesicular exosome</span></span>
<span><span class="co">#&gt; --------------------------------------</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="understanding-goseq-internals">Understanding <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em>
internals<a class="anchor" aria-label="anchor" href="#understanding-goseq-internals"></a>
</h4>
<p>The situation may arise where it is necessary for the user to perform
some of the data processing steps usually performed automatically by
<em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em>
themselves. With this in mind, it will be useful to step through the
preprocessing steps performed automatically by <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em> to
understand what is happening.</p>
<p>To start with, when <code><a href="../reference/nullp.html">nullp()</a></code> is called, <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em> uses
the genome and gene identifiers supplied to try and retrieve length
information for all genes given to the <code>genes</code> argument. To
do this, it retrieves the data from the database of gene lengths
maintained in the package <em><a href="https://bioconductor.org/packages/3.19/geneLenDataBase" class="external-link">geneLenDataBase</a></em>.<br>
This is performed by the <code><a href="../reference/getlength.html">getlength()</a></code> function in the
following way:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">len</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getlength.html">getlength</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">genes</span><span class="op">)</span>, <span class="st">"hg19"</span>, <span class="st">"ensGene"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading hg19 length data...</span></span></code></pre></div>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">len</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 22743</span></span></code></pre></div>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">genes</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 22743</span></span></code></pre></div>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">len</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1]  247 3133 1978  466 1510  954</span></span></code></pre></div>
<p>After some data cleanup, the length data and the DE data is then
passed to the <code><a href="../reference/makespline.html">makespline()</a></code> function to produce the PWF. The
<code><a href="../reference/nullp.html">nullp()</a></code> returns a data frame which has 3 columns, the
original <code>DEgenes</code> vector, the length bias data (in a column
called <code>bias.data</code>) and the PWF itself (in a column named
<code>pwf</code>). The names of the genes are also kept in this data
frame as the names of the rows. If length data could not be obtained for
a certain gene the corresponding entries in the <code>bias.data</code>
and <code>pwf</code> columns are set to <code>NA</code>.</p>
<p>Next we call the <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em>
function to determine over/under representation of GO categories amongst
DE genes. When we do this, <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em> looks
for the appropriate organism package and tries to obtain the mapping
from genes to GO categories from it. This is done using the
<code><a href="../reference/getgo.html">getgo()</a></code> function as follows:</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">go</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getgo.html">getgo</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">genes</span><span class="op">)</span>, <span class="st">"hg19"</span>, <span class="st">"ensGene"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">go</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 22743</span></span></code></pre></div>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">genes</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 22743</span></span></code></pre></div>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">go</span><span class="op">)</span></span>
<span><span class="co">#&gt; $&lt;NA&gt;</span></span>
<span><span class="co">#&gt; NULL</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $ENSG00000182463</span></span>
<span><span class="co">#&gt;  [1] "GO:0006139" "GO:0006351" "GO:0006355" "GO:0006357" "GO:0006366" "GO:0006725"</span></span>
<span><span class="co">#&gt;  [7] "GO:0006807" "GO:0008150" "GO:0008152" "GO:0009058" "GO:0009059" "GO:0009889"</span></span>
<span><span class="co">#&gt; [13] "GO:0009987" "GO:0010467" "GO:0010468" "GO:0010556" "GO:0016070" "GO:0018130"</span></span>
<span><span class="co">#&gt; [19] "GO:0019219" "GO:0019222" "GO:0019438" "GO:0031323" "GO:0031326" "GO:0032774"</span></span>
<span><span class="co">#&gt; [25] "GO:0034641" "GO:0034654" "GO:0043170" "GO:0044237" "GO:0044238" "GO:0044249"</span></span>
<span><span class="co">#&gt; [31] "GO:0044271" "GO:0046483" "GO:0050789" "GO:0050794" "GO:0051171" "GO:0051252"</span></span>
<span><span class="co">#&gt; [37] "GO:0060255" "GO:0065007" "GO:0071704" "GO:0080090" "GO:0090304" "GO:1901360"</span></span>
<span><span class="co">#&gt; [43] "GO:1901362" "GO:1901576" "GO:2001141" "GO:0000785" "GO:0005575" "GO:0005622"</span></span>
<span><span class="co">#&gt; [49] "GO:0005634" "GO:0005694" "GO:0032991" "GO:0032993" "GO:0043226" "GO:0043227"</span></span>
<span><span class="co">#&gt; [55] "GO:0043228" "GO:0043229" "GO:0043231" "GO:0043232" "GO:0110165" "GO:0000981"</span></span>
<span><span class="co">#&gt; [61] "GO:0003674" "GO:0003676" "GO:0003677" "GO:0003700" "GO:0005488" "GO:0005515"</span></span>
<span><span class="co">#&gt; [67] "GO:0036094" "GO:0043167" "GO:0043169" "GO:0046872" "GO:0097159" "GO:0140110"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $&lt;NA&gt;</span></span>
<span><span class="co">#&gt; NULL</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $&lt;NA&gt;</span></span>
<span><span class="co">#&gt; NULL</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $&lt;NA&gt;</span></span>
<span><span class="co">#&gt; NULL</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $ENSG00000125835</span></span>
<span><span class="co">#&gt;   [1] "GO:0000245" "GO:0000375" "GO:0000377" "GO:0000387" "GO:0000398" "GO:0001510"</span></span>
<span><span class="co">#&gt;   [7] "GO:0006139" "GO:0006396" "GO:0006397" "GO:0006479" "GO:0006725" "GO:0006807"</span></span>
<span><span class="co">#&gt;  [13] "GO:0008150" "GO:0008152" "GO:0008213" "GO:0008380" "GO:0009058" "GO:0009059"</span></span>
<span><span class="co">#&gt;  [19] "GO:0009451" "GO:0009987" "GO:0010467" "GO:0016043" "GO:0016070" "GO:0016071"</span></span>
<span><span class="co">#&gt;  [25] "GO:0019538" "GO:0022607" "GO:0022613" "GO:0022618" "GO:0032259" "GO:0034641"</span></span>
<span><span class="co">#&gt;  [31] "GO:0036211" "GO:0036260" "GO:0036261" "GO:0043170" "GO:0043412" "GO:0043414"</span></span>
<span><span class="co">#&gt;  [37] "GO:0043933" "GO:0044085" "GO:0044237" "GO:0044238" "GO:0044249" "GO:0046483"</span></span>
<span><span class="co">#&gt;  [43] "GO:0065003" "GO:0071704" "GO:0071826" "GO:0071840" "GO:0090304" "GO:1901360"</span></span>
<span><span class="co">#&gt;  [49] "GO:1901564" "GO:1901576" "GO:1903241" "GO:0005575" "GO:0005622" "GO:0005634"</span></span>
<span><span class="co">#&gt;  [55] "GO:0005654" "GO:0005681" "GO:0005682" "GO:0005683" "GO:0005684" "GO:0005685"</span></span>
<span><span class="co">#&gt;  [61] "GO:0005686" "GO:0005687" "GO:0005689" "GO:0005697" "GO:0005737" "GO:0005829"</span></span>
<span><span class="co">#&gt;  [67] "GO:0030532" "GO:0031974" "GO:0031981" "GO:0032991" "GO:0034708" "GO:0034709"</span></span>
<span><span class="co">#&gt;  [73] "GO:0034719" "GO:0043226" "GO:0043227" "GO:0043229" "GO:0043231" "GO:0043233"</span></span>
<span><span class="co">#&gt;  [79] "GO:0046540" "GO:0070013" "GO:0071004" "GO:0071005" "GO:0071007" "GO:0071010"</span></span>
<span><span class="co">#&gt;  [85] "GO:0071011" "GO:0071013" "GO:0071204" "GO:0097525" "GO:0097526" "GO:0110165"</span></span>
<span><span class="co">#&gt;  [91] "GO:0120114" "GO:0140513" "GO:1902494" "GO:1990234" "GO:1990904" "GO:0003674"</span></span>
<span><span class="co">#&gt;  [97] "GO:0003676" "GO:0003723" "GO:0005488" "GO:0005515" "GO:0043021" "GO:0044877"</span></span>
<span><span class="co">#&gt; [103] "GO:0070034" "GO:0070990" "GO:0071208" "GO:0097159" "GO:1990446" "GO:1990447"</span></span></code></pre></div>
<p>Note that some of the gene categories have been returned as
<code>NULL</code>. This means that a GO category could not be found in
the database for one of the genes. In the <code><a href="../reference/goseq.html">goseq()</a></code> command,
enrichment will only be calculated using genes with a GO category by
default. However, in older versions of goseq (below 1.15.2), we counted
all genes, i.e. genes with no categories still counted towards the total
number of gene outside of any single category. It is possible to switch
between these two behaviors using the <code>use_genes_without_cat</code>
flag in <code><a href="../reference/goseq.html">goseq()</a></code>.</p>
<p>The first thing the <code><a href="../reference/getgo.html">getgo()</a></code> function does is to convert
the UCSC genome/ID namings into the naming convention used by the
organism packages. This is done using two hard coded conversion vectors
that are included in the <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em>
package but usually hidden from the user.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">goseq</span><span class="fu">:::</span><span class="va">.ID_MAP</span></span>
<span><span class="co">#&gt;  knownGene    refGene    ensGene geneSymbol        sgd     plasmo       tair </span></span>
<span><span class="co">#&gt;       "eg"       "eg"  "ENSEMBL"   "SYMBOL"      "sgd"   "plasmo"     "tair"</span></span></code></pre></div>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">goseq</span><span class="fu">:::</span><span class="va">.ORG_PACKAGES</span></span>
<span><span class="co">#&gt;           anoGam      Arabidopsis           bosTau               ce </span></span>
<span><span class="co">#&gt;      "org.Ag.eg"    "org.At.tair"      "org.Bt.eg"      "org.Ce.eg" </span></span>
<span><span class="co">#&gt;           canFam               dm           danRer      E. coli K12 </span></span>
<span><span class="co">#&gt;      "org.Cf.eg"      "org.Dm.eg"      "org.Dr.eg"   "org.EcK12.eg" </span></span>
<span><span class="co">#&gt;    E. coli Sakai           galGal               hg               mm </span></span>
<span><span class="co">#&gt; "org.EcSakai.eg"      "org.Gg.eg"      "org.Hs.eg"      "org.Mm.eg" </span></span>
<span><span class="co">#&gt;           rheMac          Malaria           panTro               rn </span></span>
<span><span class="co">#&gt;     "org.Mmu.eg"  "org.Pf.plasmo"      "org.Pt.eg"      "org.Rn.eg" </span></span>
<span><span class="co">#&gt;           sacCer           susScr           xenTro </span></span>
<span><span class="co">#&gt;     "org.Sc.sgd"      "org.Ss.eg"      "org.Xl.eg"</span></span></code></pre></div>
<p>It is just as valid to run the length and GO category fetching as
separate steps and then pass the result to the <code><a href="../reference/nullp.html">nullp()</a></code> and
<code><a href="../reference/goseq.html">goseq()</a></code> functions using the <code>bias.data</code> and
<code>gene2cat</code> arguments. Thus the following two blocks of code
are equivalent:</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pwf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/nullp.html">nullp</a></span><span class="op">(</span><span class="va">genes</span>, <span class="st">"hg19"</span>, <span class="st">"ensGene"</span><span class="op">)</span></span>
<span><span class="va">go</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/goseq.html">goseq</a></span><span class="op">(</span><span class="va">pwf</span>, <span class="st">"hg19"</span>, <span class="st">"ensGene"</span><span class="op">)</span></span></code></pre></div>
<p>and</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_lengths</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getlength.html">getlength</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">genes</span><span class="op">)</span>, <span class="st">"hg19"</span>, <span class="st">"ensGene"</span><span class="op">)</span></span>
<span><span class="va">pwf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/nullp.html">nullp</a></span><span class="op">(</span><span class="va">genes</span>, bias.data <span class="op">=</span> <span class="va">gene_lengths</span><span class="op">)</span></span>
<span><span class="va">go_map</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getgo.html">getgo</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">genes</span><span class="op">)</span>, <span class="st">"hg19"</span>, <span class="st">"ensGene"</span><span class="op">)</span></span>
<span><span class="va">go</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/goseq.html">goseq</a></span><span class="op">(</span><span class="va">pwf</span>, <span class="st">"hg19"</span>, <span class="st">"ensGene"</span>, gene2cat <span class="op">=</span> <span class="va">go_map</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="kegg-pathway-analysis">KEGG pathway analysis<a class="anchor" aria-label="anchor" href="#kegg-pathway-analysis"></a>
</h3>
<p>In order to illustrate performing a category test not present in the
<em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em>
database, we perform a KEGG pathway analysis. For human, the mapping
from KEGG pathways to genes are stored in the package
<code>org.Hs.eg.db</code>, in the object <code>org.Hs.egPATH</code>. In
order to test for KEGG pathway over representation amongst DE genes, we
need to extract this information and put it in a format that <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em>
understands. Unfortunately, the <code>org.Hs.eg.db</code> package does
not contain direct mappings between ENSEMBL gene ID and KEGG pathway.
Therefore, we have to construct this map by combining the ENSEMBL
&lt;-&gt; Entrez and Entrez &lt;-&gt; KEGG mappings. This can be done
using the following code:</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get the mapping from ENSEMBL 2 Entrez</span></span>
<span><span class="va">en2eg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">org.Hs.egENSEMBL2EG</span><span class="op">)</span></span>
<span><span class="co"># Get the mapping from Entrez 2 KEGG</span></span>
<span><span class="va">eg2kegg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">org.Hs.egPATH</span><span class="op">)</span></span>
<span><span class="co"># Define a function which gets all unique KEGG IDs</span></span>
<span><span class="co"># associated with a set of Entrez IDs</span></span>
<span><span class="va">grepKEGG</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">id</span>, <span class="va">mapkeys</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">mapkeys</span><span class="op">[</span><span class="va">id</span><span class="op">]</span>, use.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="co"># Apply this function to every entry in the mapping from</span></span>
<span><span class="co"># ENSEMBL 2 Entrez to combine the two maps</span></span>
<span><span class="va">kegg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">en2eg</span>, <span class="va">grepKEGG</span>, <span class="va">eg2kegg</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">kegg</span><span class="op">)</span></span></code></pre></div>
<p>Note that this step is quite time consuming. The code written here is
not the most efficient way of producing this result, but the logic is
much clearer than faster algorithms. The source code for
<code><a href="../reference/getgo.html">getgo()</a></code> contains a more efficient routine.</p>
<p>We produce the PWF as before. Then, to perform a KEGG analysis, we
simply make use of the <code>gene2cat</code> option in
<code><a href="../reference/goseq.html">goseq()</a></code>:</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pwf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/nullp.html">nullp</a></span><span class="op">(</span><span class="va">genes</span>, <span class="st">"hg19"</span>, <span class="st">"ensGene"</span><span class="op">)</span></span>
<span><span class="va">KEGG</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/goseq.html">goseq</a></span><span class="op">(</span><span class="va">pwf</span>, gene2cat <span class="op">=</span> <span class="va">kegg</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">KEGG</span><span class="op">)</span></span></code></pre></div>
<p>Note that we do not have to tell the <code><a href="../reference/goseq.html">goseq()</a></code> function
what organism and gene ID we are using as we are manually supplying the
mapping between genes and categories.</p>
<p>KEGG analysis is shown as an illustration of how to supply your own
mapping between gene ID and category, KEGG analysis is actually natively
supported by <em><a href="https://bioconductor.org/packages/3.19/goseq" class="external-link">goseq</a></em> and
we could have performed it with the following code.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pwf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/nullp.html">nullp</a></span><span class="op">(</span><span class="va">genes</span>, <span class="st">"hg19"</span>, <span class="st">"ensGene"</span>, plot.fit <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading hg19 length data...</span></span>
<span><span class="co">#&gt; Warning in pcls(G): initial point very close to some inequality constraints</span></span></code></pre></div>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">KEGG</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/goseq.html">goseq</a></span><span class="op">(</span><span class="va">pwf</span>, <span class="st">"hg19"</span>, <span class="st">"ensGene"</span>, test.cats <span class="op">=</span> <span class="st">"KEGG"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Fetching GO annotations...</span></span>
<span><span class="co">#&gt; For 18663 genes, we could not find any categories. These genes will be excluded.</span></span>
<span><span class="co">#&gt; To force their use, please run with use_genes_without_cat=TRUE (see documentation).</span></span>
<span><span class="co">#&gt; This was the default behavior for version 1.15.1 and earlier.</span></span>
<span><span class="co">#&gt; Calculating the p-values...</span></span></code></pre></div>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">KEGG</span><span class="op">)</span></span>
<span><span class="co">#&gt;     category over_represented_pvalue under_represented_pvalue numDEInCat numInCat</span></span>
<span><span class="co">#&gt; 88     03010            6.317053e-06                0.9999980         29       87</span></span>
<span><span class="co">#&gt; 77     00900            2.391514e-04                0.9999710         10       15</span></span>
<span><span class="co">#&gt; 113    04115            8.164693e-04                0.9996835         26       64</span></span>
<span><span class="co">#&gt; 175    04964            2.150726e-03                0.9995925         10       17</span></span>
<span><span class="co">#&gt; 27     00330            3.668792e-03                0.9986594         18       44</span></span>
<span><span class="co">#&gt; 20     00250            5.200138e-03                0.9984357         13       28</span></span></code></pre></div>
<p>Noting that this time it was necessary to tell the
<code><a href="../reference/goseq.html">goseq()</a></code> function that we are using hg19 and ENSEMBL gene
ID, as the function needs this information to automatically construct
the mapping from geneid to KEGG pathway.</p>
</div>
<div class="section level3">
<h3 id="extracting-mappings-from-organism-packages">Extracting mappings from organism packages<a class="anchor" aria-label="anchor" href="#extracting-mappings-from-organism-packages"></a>
</h3>
<p>If you know that the information mapping gene ID to your categories
of interest is contained in the organism packages, but
<code><a href="../reference/goseq.html">goseq()</a></code> fails to fetch it automatically, you may want to
extract it yourself and then pass it to the <code><a href="../reference/goseq.html">goseq()</a></code>
function using the <code>gene2cat</code> argument. This is done in
exactly the same way as extracting the KEGG to ENSEMBL mappings in the
section “KEGG pathway analysis” above. This example is actually the
worst case, where it is necessary to combine two mappings to get the
desired list. If we had instead wanted the association between Entrez
gene IDs and KEGG pathways, the following code would have been
sufficient:</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">kegg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">org.Hs.egPATH</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">kegg</span><span class="op">)</span></span>
<span><span class="co">#&gt; $`1`</span></span>
<span><span class="co">#&gt; [1] NA</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`2`</span></span>
<span><span class="co">#&gt; [1] "04610"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`3`</span></span>
<span><span class="co">#&gt; [1] NA</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`9`</span></span>
<span><span class="co">#&gt; [1] "00232" "00983" "01100"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`10`</span></span>
<span><span class="co">#&gt; [1] "00232" "00983" "01100"</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $`11`</span></span>
<span><span class="co">#&gt; [1] NA</span></span></code></pre></div>
<p>A note on fetching GO mappings from the organism.<br>
The data structure of GO is a directed acyclic graph. This means that in
addition to each GO category being associated with a set of genes, it
may also have children that are associated to other genes. It is
important to use the <code>org.Hs.egGO2ALLEGS</code> and NOT the
<code>org.Hs.egGO</code> object to create the mapping between GO
categories and gene identifiers, as the latter does not include the
links to genes arising from “child” GO categories. Thank you to
Christopher Fjell for pointing this out.</p>
</div>
<div class="section level3">
<h3 id="correcting-for-other-biases">Correcting for other biases<a class="anchor" aria-label="anchor" href="#correcting-for-other-biases"></a>
</h3>
<p>It is possible that in some circumstances you will wish to correct
not just for length bias, but for the total number of counts. This can
make sense because power to detect DE depends on the total number of
counts a gene receives, which is the product of gene length and gene
expression. So correcting for read count bias will compensate for all
biases, known and unknown, in power to detect DE. On the other hand, it
will also remove bias resulting from differences in expression level,
which may not be desirable.</p>
<p>Correcting for count bias will produce a different PWF. Therefore, we
need to tell <code><a href="../reference/goseq.html">goseq()</a></code> about the data on which the fraction
DE depends when calculating the PWF using the <code><a href="../reference/nullp.html">nullp()</a></code>
function.<br>
We then simply pass the result to <code><a href="../reference/goseq.html">goseq()</a></code> as usual.</p>
<p>So, in order to tell <code><a href="../reference/goseq.html">goseq()</a></code> to correct for read count
bias instead of length bias, all you need to do is supply a numeric
vector, containing the number of counts for each gene to
<code><a href="../reference/nullp.html">nullp()</a></code>.</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">countbias</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="va">counts</span><span class="op">)</span> <span class="op">!=</span> <span class="fl">0</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">countbias</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 22743</span></span></code></pre></div>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">genes</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 22743</span></span></code></pre></div>
<p>To use the count bias when doing GO analysis, simply pass this vector
to <code><a href="../reference/nullp.html">nullp()</a></code> using the <code>bias.data</code> option. Note
that we have to supply <code>hg19</code> and <code>ensGene</code> to
<code><a href="../reference/goseq.html">goseq()</a></code> as it is not used by <code><a href="../reference/nullp.html">nullp()</a></code> and hence
not in the <code>pwf.counts</code> object.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pwf.counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/nullp.html">nullp</a></span><span class="op">(</span><span class="va">genes</span>, bias.data <span class="op">=</span> <span class="va">countbias</span>, plot.fit <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in pcls(G): initial point very close to some inequality constraints</span></span></code></pre></div>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">GO.counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/goseq.html">goseq</a></span><span class="op">(</span><span class="va">pwf.counts</span>, <span class="st">"hg19"</span>, <span class="st">"ensGene"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Fetching GO annotations...</span></span>
<span><span class="co">#&gt; For 9343 genes, we could not find any categories. These genes will be excluded.</span></span>
<span><span class="co">#&gt; To force their use, please run with use_genes_without_cat=TRUE (see documentation).</span></span>
<span><span class="co">#&gt; This was the default behavior for version 1.15.1 and earlier.</span></span>
<span><span class="co">#&gt; Calculating the p-values...</span></span>
<span><span class="co">#&gt; 'select()' returned 1:1 mapping between keys and columns</span></span></code></pre></div>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">GO.counts</span><span class="op">)</span></span>
<span><span class="co">#&gt;         category over_represented_pvalue under_represented_pvalue numDEInCat</span></span>
<span><span class="co">#&gt; 15209 GO:0071944            9.653696e-16                        1        852</span></span>
<span><span class="co">#&gt; 2590  GO:0005886            5.514780e-14                        1        791</span></span>
<span><span class="co">#&gt; 6631  GO:0023052            3.072878e-09                        1        961</span></span>
<span><span class="co">#&gt; 3352  GO:0007154            3.721183e-09                        1        974</span></span>
<span><span class="co">#&gt; 12092 GO:0048731            1.204307e-08                        1        666</span></span>
<span><span class="co">#&gt; 3705  GO:0008283            1.679259e-08                        1        345</span></span>
<span><span class="co">#&gt;       numInCat                          term ontology</span></span>
<span><span class="co">#&gt; 15209     3638                cell periphery       CC</span></span>
<span><span class="co">#&gt; 2590      3369               plasma membrane       CC</span></span>
<span><span class="co">#&gt; 6631      4171                     signaling       BP</span></span>
<span><span class="co">#&gt; 3352      4237            cell communication       BP</span></span>
<span><span class="co">#&gt; 12092     2769            system development       BP</span></span>
<span><span class="co">#&gt; 3705      1338 cell population proliferation       BP</span></span></code></pre></div>
<p>Note that if you want to correct for length bias, but your
organism/gene identifier is not natively supported, then you need to
follow the same procedure as above, only the numeric vector supplied
will contain each gene’s length instead of its number of reads.</p>
</div>
</div>
<div class="section level2">
<h2 id="acknowledgments">Acknowledgments<a class="anchor" aria-label="anchor" href="#acknowledgments"></a>
</h2>
<p>Christopher Fjell for a series of bug fixes and pointing out the
difference between the <code>egGO</code> and <code>egGO2ALLEGS</code>
objects in the organism packages.</p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<p>This vignette was built on:</p>
<pre><code><span><span class="co">#&gt; R version 4.4.0 (2024-04-24)</span></span>
<span><span class="co">#&gt; Platform: x86_64-apple-darwin20</span></span>
<span><span class="co">#&gt; Running under: macOS Monterey 12.7.1</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.4-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Europe/Berlin</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] GO.db_3.19.1           org.Hs.eg.db_3.19.1    AnnotationDbi_1.66.0  </span></span>
<span><span class="co">#&gt;  [4] IRanges_2.38.0         S4Vectors_0.42.0       Biobase_2.64.0        </span></span>
<span><span class="co">#&gt;  [7] BiocGenerics_0.50.0    edgeR_4.2.0            limma_3.60.2          </span></span>
<span><span class="co">#&gt; [10] goseq_1.57.3           geneLenDataBase_1.41.2 BiasedUrn_2.0.11      </span></span>
<span><span class="co">#&gt; [13] BiocStyle_2.32.0      </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] DBI_1.2.3                   bitops_1.0-7               </span></span>
<span><span class="co">#&gt;   [3] httr2_1.0.1                 biomaRt_2.60.0             </span></span>
<span><span class="co">#&gt;   [5] rlang_1.1.4                 magrittr_2.0.3             </span></span>
<span><span class="co">#&gt;   [7] matrixStats_1.3.0           compiler_4.4.0             </span></span>
<span><span class="co">#&gt;   [9] RSQLite_2.3.7               mgcv_1.9-1                 </span></span>
<span><span class="co">#&gt;  [11] GenomicFeatures_1.57.0      png_0.1-8                  </span></span>
<span><span class="co">#&gt;  [13] systemfonts_1.1.0           vctrs_0.6.5                </span></span>
<span><span class="co">#&gt;  [15] txdbmaker_1.0.0             stringr_1.5.1              </span></span>
<span><span class="co">#&gt;  [17] pkgconfig_2.0.3             crayon_1.5.2               </span></span>
<span><span class="co">#&gt;  [19] fastmap_1.2.0               dbplyr_2.5.0               </span></span>
<span><span class="co">#&gt;  [21] XVector_0.44.0              utf8_1.2.4                 </span></span>
<span><span class="co">#&gt;  [23] Rsamtools_2.20.0            rmarkdown_2.27             </span></span>
<span><span class="co">#&gt;  [25] UCSC.utils_1.0.0            ragg_1.3.2                 </span></span>
<span><span class="co">#&gt;  [27] purrr_1.0.2                 bit_4.0.5                  </span></span>
<span><span class="co">#&gt;  [29] xfun_0.44                   zlibbioc_1.50.0            </span></span>
<span><span class="co">#&gt;  [31] cachem_1.1.0                GenomeInfoDb_1.40.1        </span></span>
<span><span class="co">#&gt;  [33] jsonlite_1.8.8              progress_1.2.3             </span></span>
<span><span class="co">#&gt;  [35] blob_1.2.4                  highr_0.11                 </span></span>
<span><span class="co">#&gt;  [37] DelayedArray_0.30.1         BiocParallel_1.38.0        </span></span>
<span><span class="co">#&gt;  [39] parallel_4.4.0              prettyunits_1.2.0          </span></span>
<span><span class="co">#&gt;  [41] R6_2.5.1                    bslib_0.7.0                </span></span>
<span><span class="co">#&gt;  [43] stringi_1.8.4               rtracklayer_1.64.0         </span></span>
<span><span class="co">#&gt;  [45] GenomicRanges_1.56.0        jquerylib_0.1.4            </span></span>
<span><span class="co">#&gt;  [47] Rcpp_1.0.12                 bookdown_0.39              </span></span>
<span><span class="co">#&gt;  [49] SummarizedExperiment_1.34.0 knitr_1.47                 </span></span>
<span><span class="co">#&gt;  [51] Matrix_1.7-0                splines_4.4.0              </span></span>
<span><span class="co">#&gt;  [53] tidyselect_1.2.1            rstudioapi_0.16.0          </span></span>
<span><span class="co">#&gt;  [55] abind_1.4-5                 yaml_2.3.8                 </span></span>
<span><span class="co">#&gt;  [57] codetools_0.2-20            curl_5.2.1                 </span></span>
<span><span class="co">#&gt;  [59] lattice_0.22-6              tibble_3.2.1               </span></span>
<span><span class="co">#&gt;  [61] KEGGREST_1.44.0             evaluate_0.24.0            </span></span>
<span><span class="co">#&gt;  [63] desc_1.4.3                  BiocFileCache_2.12.0       </span></span>
<span><span class="co">#&gt;  [65] xml2_1.3.6                  Biostrings_2.72.1          </span></span>
<span><span class="co">#&gt;  [67] pillar_1.9.0                BiocManager_1.30.23        </span></span>
<span><span class="co">#&gt;  [69] filelock_1.0.3              MatrixGenerics_1.16.0      </span></span>
<span><span class="co">#&gt;  [71] generics_0.1.3              RCurl_1.98-1.14            </span></span>
<span><span class="co">#&gt;  [73] hms_1.1.3                   glue_1.7.0                 </span></span>
<span><span class="co">#&gt;  [75] tools_4.4.0                 BiocIO_1.14.0              </span></span>
<span><span class="co">#&gt;  [77] locfit_1.5-9.9              GenomicAlignments_1.40.0   </span></span>
<span><span class="co">#&gt;  [79] fs_1.6.4                    XML_3.99-0.16.1            </span></span>
<span><span class="co">#&gt;  [81] grid_4.4.0                  nlme_3.1-165               </span></span>
<span><span class="co">#&gt;  [83] GenomeInfoDbData_1.2.12     restfulr_0.0.15            </span></span>
<span><span class="co">#&gt;  [85] cli_3.6.2                   rappdirs_0.3.3             </span></span>
<span><span class="co">#&gt;  [87] textshaping_0.4.0           fansi_1.0.6                </span></span>
<span><span class="co">#&gt;  [89] S4Arrays_1.4.1              dplyr_1.1.4                </span></span>
<span><span class="co">#&gt;  [91] sass_0.4.9                  digest_0.6.35              </span></span>
<span><span class="co">#&gt;  [93] SparseArray_1.4.8           rjson_0.2.21               </span></span>
<span><span class="co">#&gt;  [95] htmlwidgets_1.6.4           memoise_2.0.1              </span></span>
<span><span class="co">#&gt;  [97] htmltools_0.5.8.1           pkgdown_2.0.9              </span></span>
<span><span class="co">#&gt;  [99] lifecycle_1.0.4             httr_1.4.7                 </span></span>
<span><span class="co">#&gt; [101] statmod_1.5.0               bit64_4.0.5</span></span></code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-Benjamini95" class="csl-entry">
Benjamini, Y., and Y. Hochberg. 1995. <span>“<a href="">Controlling the
False Discovery Rate: A Practical and Powerful Approach to Multiple
Testing</a>.”</span> <em>Journal of the Royal Statistical Society:
Series B</em> 57: 289–300.
</div>
<div id="ref-Li:2008p113" class="csl-entry">
Li, Hairi, Michael T Lovci, Young-Soo Kwon, Michael G Rosenfeld,
Xiang-Dong Fu, and Gene W Yeo. 2008. <span>“Determination of Tag Density
Required for Digital Transcriptome Analysis: Application to an
Androgen-Sensitive Prostate Cancer Model.”</span> <em>Proc Natl Acad Sci
USA</em> 105 (51): 20179–84. <a href="https://doi.org/10.1073/pnas.0807121105" class="external-link">https://doi.org/10.1073/pnas.0807121105</a>.
</div>
<div id="ref-Oshlack:2009p149" class="csl-entry">
Oshlack, Alicia, and Matthew J Wakefield. 2009. <span>“Transcript Length
Bias in <span>RNA</span>-Seq Data Confounds Systems Biology.”</span>
<em>Biol Direct</em> 4 (January): 14. <a href="https://doi.org/10.1186/1745-6150-4-14" class="external-link">https://doi.org/10.1186/1745-6150-4-14</a>.
</div>
<div id="ref-Robinson10" class="csl-entry">
Robinson, M. D., D. J. McCarthy, and G. K. Smyth. 2010.
<span>“Edge<span>R</span>: A Bioconductor Package for Differential
Expression Analysis of Digital Gene Expression Data.”</span>
<em>Bioinformatics</em> 26 (1): 139–40. <a href="https://doi.org/10.1093/bioinformatics/btp616" class="external-link">https://doi.org/10.1093/bioinformatics/btp616</a>.
</div>
<div id="ref-Robinson07" class="csl-entry">
Robinson, M. D., and G. K. Smyth. 2007. <span>“<a href="">Moderated
Statistical Tests for Assessing Differences in Tag
Abundance</a>.”</span> <em>Bioinformatics</em> 23 (21): 2881–87.
</div>
<div id="ref-Robinson08" class="csl-entry">
———. 2008. <span>“<a href="">Small-Sample Estimation of Negative
Binomial Dispersion, with Applications to SAGE Data</a>.”</span>
<em>Biostatistics</em> 9 (2): 321–32.
</div>
<div id="ref-Young10" class="csl-entry">
Young, M. D., M. J. Wakefield, G. K. Smyth, and A. Oshlack. 2010.
<span>“Gene Ontology Analysis for <span>RNA</span>-Seq: Accounting for
Selection Bias.”</span> <em>Genome Biology</em> 11 (February): R14.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Matthew Young, Nadia Davidson, Federico Marini.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
